// @aureole/core/esm2015/lib/services/caption.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysCaption } from '../entities/sys_caption.entity';
let CaptionService = class CaptionService {
    /**
     * @param {?} captionRepository
     */
    constructor(captionRepository) {
        this.captionRepository = captionRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCaptions(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.captionRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCaption(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.captionRepository.find({
                company: context.company,
                lang: context.lang,
                pid: context.pid,
                type: context.type,
                code: context.code
            });
        });
    }
};
CaptionService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysCaption)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], CaptionService);
export { CaptionService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    CaptionService.prototype.captionRepository;
}
