// @aureole/core/esm2015/lib/services/company.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysCompany } from '../entities/sys_company.entity';
let CompanyService = class CompanyService {
    /**
     * @param {?} companyRepository
     */
    constructor(companyRepository) {
        this.companyRepository = companyRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCompanies(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.companyRepository.find({
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCompany(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.companyRepository.find({
                lang: context.lang,
                code: context.code
            });
        });
    }
};
CompanyService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysCompany)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], CompanyService);
export { CompanyService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    CompanyService.prototype.companyRepository;
}
