// @aureole/core/esm2015/lib/services/define-class.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { SysDefineClass } from '../entities/sys_define_class.entity';
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
let DefineClassService = class DefineClassService {
    /**
     * @param {?} defineClassRepository
     */
    constructor(defineClassRepository) {
        this.defineClassRepository = defineClassRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineClasses(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineClassRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineClass(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineClassRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
};
DefineClassService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineClass)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineClassService);
export { DefineClassService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineClassService.prototype.defineClassRepository;
}
