// @aureole/core/esm2015/lib/services/define-search.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineSearch } from '../entities/sys_define_search.entity';
let DefineSearchService = class DefineSearchService {
    /**
     * @param {?} defineSearchRepository
     */
    constructor(defineSearchRepository) {
        this.defineSearchRepository = defineSearchRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineSearch(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineSearchRepository.find({
                company: context.company,
                pid: context.pid,
                search_id: context.search_id
            });
        });
    }
};
DefineSearchService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineSearch)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineSearchService);
export { DefineSearchService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineSearchService.prototype.defineSearchRepository;
}
