// @aureole/core/esm2015/lib/services/define-input-grid.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineInputGrid } from '../entities/sys_define_input_grid.entity';
let DefineInputGridService = class DefineInputGridService {
    /**
     * @param {?} defineInputGridRepository
     */
    constructor(defineInputGridRepository) {
        this.defineInputGridRepository = defineInputGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineInputGrid(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineInputGridRepository.find({
                company: context.company,
                pid: context.pid,
                input_grid_id: context.input_grid_id
            });
        });
    }
};
DefineInputGridService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineInputGrid)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineInputGridService);
export { DefineInputGridService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineInputGridService.prototype.defineInputGridRepository;
}
