// @aureole/core/esm2015/lib/services/define-mapping-control.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineMappingControl } from '../entities/sys_define_mapping_control.entity';
let DefineMappingControlService = class DefineMappingControlService {
    /**
     * @param {?} defineMappingControlRepository
     */
    constructor(defineMappingControlRepository) {
        this.defineMappingControlRepository = defineMappingControlRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineMappingControls(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineMappingControlRepository.find({
                company: context.company,
                pid: context.pid
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineMappingControl(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineMappingControlRepository.find({
                company: context.company,
                pid: context.pid,
                item_no: context.item_no
            });
        });
    }
};
DefineMappingControlService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineMappingControl)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineMappingControlService);
export { DefineMappingControlService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineMappingControlService.prototype.defineMappingControlRepository;
}
