// @aureole/core/esm2015/lib/services/define-group.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineGroup } from '../entities/sys_define_group.entity';
let DefineGroupService = class DefineGroupService {
    /**
     * @param {?} defineGroupRepository
     */
    constructor(defineGroupRepository) {
        this.defineGroupRepository = defineGroupRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineGroups(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineGroupRepository.find({
                company: context.company,
                pid: context.pid
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineGroup(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineGroupRepository.find({
                company: context.company,
                pid: context.pid,
                group_id: context.group_id
            });
        });
    }
};
DefineGroupService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineGroup)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineGroupService);
export { DefineGroupService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineGroupService.prototype.defineGroupRepository;
}
