// @aureole/core/esm2015/lib/services/define-input.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineInput } from '../entities/sys_define_input.entity';
let DefineInputService = class DefineInputService {
    /**
     * @param {?} defineInputRepository
     */
    constructor(defineInputRepository) {
        this.defineInputRepository = defineInputRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineInput(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineInputRepository.find({
                company: context.company,
                pid: context.pid,
                input_id: context.input_id
            });
        });
    }
};
DefineInputService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineInput)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineInputService);
export { DefineInputService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineInputService.prototype.defineInputRepository;
}
