// @aureole/core/esm2015/lib/services/define-display.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineDisplay } from '../entities/sys_define_display.entity';
let DefineDisplayService = class DefineDisplayService {
    /**
     * @param {?} defineDisplayRepository
     */
    constructor(defineDisplayRepository) {
        this.defineDisplayRepository = defineDisplayRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineDisplays(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineDisplayRepository.find({
                company: context.company,
                emp_id: context.emp_id,
                pid: context.pid,
                group_id: context.group_id
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineDisplay(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.defineDisplayRepository.find({
                company: context.company,
                emp_id: context.emp_id,
                pid: context.pid,
                group_id: context.group_id,
                item_no: context.item_no
            });
        });
    }
};
DefineDisplayService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDefineDisplay)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DefineDisplayService);
export { DefineDisplayService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineDisplayService.prototype.defineDisplayRepository;
}
