// @aureole/core/esm2015/lib/services/department.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDepartment } from '../entities/sys_department.entity';
let DepartmentService = class DepartmentService {
    /**
     * @param {?} departmentRepository
     */
    constructor(departmentRepository) {
        this.departmentRepository = departmentRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDepartments(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.departmentRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDepartment(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.departmentRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
};
DepartmentService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysDepartment)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], DepartmentService);
export { DepartmentService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DepartmentService.prototype.departmentRepository;
}
