// @aureole/core/esm2015/lib/services/employee.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysEmployee } from '../entities/sys_employee.entity';
let EmployeeService = class EmployeeService {
    /**
     * @param {?} employeeRepository
     */
    constructor(employeeRepository) {
        this.employeeRepository = employeeRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getEmployees(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let params = new Object();
            /** @type {?} */
            let list = [];
            if (context.name !== undefined) {
                params['name'] = context.name;
            }
            if (context.activeFlag !== undefined) {
                params['active_flag'] = context.activeFlag;
            }
            if (context.address1 !== undefined) {
                params['address1'] = context.address1;
            }
            if (context.address2 !== undefined) {
                params['address2'] = context.address2;
            }
            if (context.tel1 !== undefined) {
                params['tel1'] = context.tel1;
            }
            if (context.tel2 !== undefined) {
                params['tel2'] = context.tel2;
            }
            if (context.email !== undefined) {
                params['email'] = context.email;
            }
            if (context.sortNo !== undefined) {
                params['sort_no'] = context.sortNo;
            }
            if (context.changeCount !== undefined) {
                params['change_count'] = context.changeCount;
            }
            if (context.dataFlag !== undefined) {
                params['data_flag'] = context.dataFlag;
            }
            /** @type {?} */
            let result = yield this.employeeRepository.find(Object.assign({ company: context.company, lang: context.lang }, params));
            list = [...result];
            // NOTE Emp code
            if (context.codeFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.code === context.codeFrom;
                }));
                if (index >= 0) {
                    list = list.slice(index);
                }
            }
            if (context.codeTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.code === context.codeTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE create_emp_id
            if (context.createEmpFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_emp_id === context.createEmpFrom;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.createEmpTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_emp_id === context.createEmpTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE create_datetime
            if (context.createDatetimeFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_datetime === new Date(context.createDatetimeFrom);
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.createDatetimeTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_datetime === context.createDatetimeTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE change_emp_id
            if (context.changeEmpFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_emp_id === context.changeEmpFrom;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.changeEmpTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_emp_id === context.changeEmpTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE change_datetime
            if (context.changeDatetimeFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_datetime === context.changeDatetimeFrom;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.changeDatetimeTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_datetime === context.changeDatetimeTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            return list;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getEmployee(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.employeeRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    addEmployee(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            if (context.code !== '') {
                /** @type {?} */
                let result = yield this.employeeRepository.find({
                    company: context.company,
                    code: context.code
                });
                if (result.length <= 0) {
                    /** @type {?} */
                    let contextVi = Object.assign({}, context);
                    contextVi.lang = 'vi-VN';
                    /** @type {?} */
                    let contextEn = Object.assign({}, context);
                    contextEn.lang = 'en-US';
                    /** @type {?} */
                    let contextJa = Object.assign({}, context);
                    contextJa.lang = 'ja-JP';
                    yield this.employeeRepository.save(contextVi);
                    yield this.employeeRepository.save(contextEn);
                    yield this.employeeRepository.save(contextJa);
                    return true;
                }
                return false;
            }
            return false;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    updateEmployee(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            if (context.code !== '') {
                /** @type {?} */
                let result = yield this.employeeRepository.find({
                    company: context.company,
                    code: context.code
                });
                if (result.length > 0) {
                    context.change_count++;
                    context.change_datetime = null;
                    /** @type {?} */
                    let contextVi = Object.assign({}, context);
                    contextVi.lang = 'vi-VN';
                    /** @type {?} */
                    let contextEn = Object.assign({}, context);
                    contextEn.lang = 'en-US';
                    /** @type {?} */
                    let contextJa = Object.assign({}, context);
                    contextJa.lang = 'ja-JP';
                    yield this.employeeRepository.update([contextVi.company, contextVi.lang, contextVi.code], contextVi);
                    yield this.employeeRepository.update([contextEn.company, contextEn.lang, contextEn.code], contextEn);
                    yield this.employeeRepository.update([contextJa.company, contextJa.lang, contextJa.code], contextJa);
                    return true;
                }
                return false;
            }
            return false;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    deleteEmployee(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let result = yield this.employeeRepository.delete({ code: context });
            if (result.affected > 0) {
                return true;
            }
            return false;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    deleteEmployees(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let result = yield this.employeeRepository.delete(context);
            if (result.affected > 0) {
                return true;
            }
            return false;
        });
    }
};
EmployeeService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysEmployee)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], EmployeeService);
export { EmployeeService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    EmployeeService.prototype.employeeRepository;
}
