// @aureole/core/esm2015/lib/services/lang.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysLang } from '../entities/sys_lang.entity';
let LangService = class LangService {
    /**
     * @param {?} langRepository
     */
    constructor(langRepository) {
        this.langRepository = langRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getLangs(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.langRepository.find({
                company: context.company
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getLang(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.langRepository.find({
                company: context.company,
                code: context.code
            });
        });
    }
};
LangService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysLang)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], LangService);
export { LangService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    LangService.prototype.langRepository;
}
