// @aureole/core/esm2015/lib/services/master.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysMaster } from '../entities/sys_master.entity';
let MasterService = class MasterService {
    /**
     * @param {?} masterRepository
     */
    constructor(masterRepository) {
        this.masterRepository = masterRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMasters(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.masterRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMaster(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.masterRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code,
                code: context.code
            });
        });
    }
};
MasterService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysMaster)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], MasterService);
export { MasterService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MasterService.prototype.masterRepository;
}
