// @aureole/core/esm2015/lib/services/master-detail.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysMasterDetail } from '../entities/sys_master_detail.entity';
let MasterDetailService = class MasterDetailService {
    /**
     * @param {?} masterDetailRepository
     */
    constructor(masterDetailRepository) {
        this.masterDetailRepository = masterDetailRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMasterDetails(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.masterDetailRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code,
                master_code: context.master_code
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMasterDetail(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.masterDetailRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code,
                master_code: context.master_code,
                code: context.code
            });
        });
    }
};
MasterDetailService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysMasterDetail)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], MasterDetailService);
export { MasterDetailService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MasterDetailService.prototype.masterDetailRepository;
}
