// @aureole/core/esm2015/lib/services/message.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysMessage } from '../entities/sys_message.entity';
let MessageService = class MessageService {
    /**
     * @param {?} messageRepository
     */
    constructor(messageRepository) {
        this.messageRepository = messageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMessage(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.messageRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
};
MessageService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysMessage)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], MessageService);
export { MessageService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MessageService.prototype.messageRepository;
}
