// @aureole/core/esm2015/lib/services/module.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysModule } from '../entities/sys_module.entity';
let ModuleService = class ModuleService {
    /**
     * @param {?} moduleRepository
     */
    constructor(moduleRepository) {
        this.moduleRepository = moduleRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getModules(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.moduleRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getModule(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.moduleRepository.find({
                company: context.company,
                lang: context.lang,
                mid: context.mid
            });
        });
    }
};
ModuleService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysModule)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], ModuleService);
export { ModuleService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ModuleService.prototype.moduleRepository;
}
