// @aureole/core/esm2015/lib/services/page.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysPage } from '../entities/sys_page.entity';
let PageService = class PageService {
    /**
     * @param {?} pageRepository
     */
    constructor(pageRepository) {
        this.pageRepository = pageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getPages(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.pageRepository.find({
                company: context.company,
                lang: context.lang,
                mid: context.mid
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getPage(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return yield this.pageRepository.find({
                company: context.company,
                lang: context.lang,
                mid: context.mid,
                pid: context.pid
            });
        });
    }
};
PageService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__param(0, InjectRepository(SysPage)),
    tslib_1.__metadata("design:paramtypes", [Repository])
], PageService);
export { PageService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    PageService.prototype.pageRepository;
}
