// @nrwl/builders/src/web/build/web-build.builder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular-devkit/core");
var build_webpack_1 = require("@angular-devkit/build-webpack");
var fs_1 = require("fs");
var path_1 = require("path");
var normalize_1 = require("../../utils/normalize");
var web_config_1 = require("../../utils/webpack/web.config");
var BuildWebBuilder = /** @class */ (function () {
    function BuildWebBuilder(context) {
        this.context = context;
        this.webpackBuilder = new build_webpack_1.WebpackBuilder(this.context);
        this.root = core_1.getSystemPath(this.context.workspace.root);
    }
    BuildWebBuilder.prototype.run = function (builderConfig) {
        var _this = this;
        var options = normalize_1.normalizeWebBuildOptions(builderConfig.options, this.root, builderConfig.sourceRoot);
        var config = web_config_1.getWebConfig(options, this.context.logger);
        if (options.webpackConfig) {
            config = require(options.webpackConfig)(config, {
                options: options,
                configuration: this.context.targetSpecifier.configuration
            });
        }
        return this.webpackBuilder.runWebpack(config, function (stats) {
            if (options.statsJson) {
                fs_1.writeFileSync(path_1.resolve(_this.root, options.outputPath, 'stats.json'), JSON.stringify(stats.toJson(), null, 2));
            }
            _this.context.logger.info(stats.toString());
        });
    };
    return BuildWebBuilder;
}());
exports.default = BuildWebBuilder;
