// @aureole/core/bundles/ait-ait-core.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/router'), require('typeorm'), require('@nestjs/common'), require('@nestjs/typeorm')) :
    typeof define === 'function' && define.amd ? define('@ait/ait-core', ['exports', '@angular/common', '@angular/core', '@angular/router', 'typeorm', '@nestjs/common', '@nestjs/typeorm'], factory) :
    (global = global || self, factory((global.ait = global.ait || {}, global.ait['ait-core'] = {}), global.ng.common, global.ng.core, global.ng.router, global.typeorm, global.common$1, global.typeorm$1));
}(this, function (exports, common, core, router, typeorm, common$1, typeorm$1) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var dbConfig = {
        type: 'postgres',
        host: '192.168.136.12',
        port: 1102,
        username: 'ait',
        password: 'ait',
        database: 'ait',
        schema: 'ait_pm_dev',
        synchronize: true
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysCaption = /** @class */ (function () {
        function SysCaption() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "type", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "code", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "name", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysCaption.prototype, "data_flag", void 0);
        SysCaption = __decorate([
            typeorm.Entity()
        ], SysCaption);
        return SysCaption;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysCompany = /** @class */ (function () {
        function SysCompany() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "address1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "address2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "tel1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "tel2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "email", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysCompany.prototype, "data_flag", void 0);
        SysCompany = __decorate([
            typeorm.Entity()
        ], SysCompany);
        return SysCompany;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineClass = /** @class */ (function () {
        function SysDefineClass() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "remark", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "table_ref", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineClass.prototype, "data_flag", void 0);
        SysDefineClass = __decorate([
            typeorm.Entity()
        ], SysDefineClass);
        return SysDefineClass;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineDisplay = /** @class */ (function () {
        function SysDefineDisplay() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "emp_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "group_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "item_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "disp_no", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "item_disp", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "width", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "height", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "row_start", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "row_end", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "col_start", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "col_end", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "color_bg", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "color_text", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "class", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "stype", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineDisplay.prototype, "data_flag", void 0);
        SysDefineDisplay = __decorate([
            typeorm.Entity()
        ], SysDefineDisplay);
        return SysDefineDisplay;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineGrid = /** @class */ (function () {
        function SysDefineGrid() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "grid_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "item_no", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "data_type", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 100 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "data_format", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "cap_type", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "cap_code", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineGrid.prototype, "data_flag", void 0);
        SysDefineGrid = __decorate([
            typeorm.Entity()
        ], SysDefineGrid);
        return SysDefineGrid;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineGroup = /** @class */ (function () {
        function SysDefineGroup() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "group_id", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "cap_type", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "cap_code", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineGroup.prototype, "data_flag", void 0);
        SysDefineGroup = __decorate([
            typeorm.Entity()
        ], SysDefineGroup);
        return SysDefineGroup;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineInput = /** @class */ (function () {
        function SysDefineInput() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "input_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "item_no", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "input_type", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "cap_type", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "cap_code", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "default_value", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineInput.prototype, "data_flag", void 0);
        SysDefineInput = __decorate([
            typeorm.Entity()
        ], SysDefineInput);
        return SysDefineInput;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineInputGrid = /** @class */ (function () {
        function SysDefineInputGrid() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "input_grid_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "item_no", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "input_type", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "cap_type", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "cap_code", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "default_value", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineInputGrid.prototype, "data_flag", void 0);
        SysDefineInputGrid = __decorate([
            typeorm.Entity()
        ], SysDefineInputGrid);
        return SysDefineInputGrid;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineMapping = /** @class */ (function () {
        function SysDefineMapping() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "mapping_group_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "mapping_item_no", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 250 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "mapping_item_field", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "mapping_table_name", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "mapping_column_name", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "mapping_data_type", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineMapping.prototype, "data_flag", void 0);
        SysDefineMapping = __decorate([
            typeorm.Entity()
        ], SysDefineMapping);
        return SysDefineMapping;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineMappingControl = /** @class */ (function () {
        function SysDefineMappingControl() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "item_no", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "control_type", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_table", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column0", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column3", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column4", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column5", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column6", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column7", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column8", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "mapping_column9", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineMappingControl.prototype, "data_flag", void 0);
        SysDefineMappingControl = __decorate([
            typeorm.Entity()
        ], SysDefineMappingControl);
        return SysDefineMappingControl;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineMappingFunction = /** @class */ (function () {
        function SysDefineMappingFunction() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "screen_type", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "function_type", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineMappingFunction.prototype, "data_flag", void 0);
        SysDefineMappingFunction = __decorate([
            typeorm.Entity()
        ], SysDefineMappingFunction);
        return SysDefineMappingFunction;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDefineSearch = /** @class */ (function () {
        function SysDefineSearch() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "pid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "search_id", void 0);
        __decorate([
            typeorm.PrimaryColumn('integer'),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "item_no", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "search_type", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "cap_type", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "cap_code", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "default_value", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDefineSearch.prototype, "data_flag", void 0);
        SysDefineSearch = __decorate([
            typeorm.Entity()
        ], SysDefineSearch);
        return SysDefineSearch;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysDepartment = /** @class */ (function () {
        function SysDepartment() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "address1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "address2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "tel1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "tel2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "email", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysDepartment.prototype, "data_flag", void 0);
        SysDepartment = __decorate([
            typeorm.Entity()
        ], SysDepartment);
        return SysDepartment;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysEmployee = /** @class */ (function () {
        function SysEmployee() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "address1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 250 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "address2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "tel1", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "tel2", void 0);
        __decorate([
            typeorm.Column('character varying', { nullable: true, length: 50 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "email", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysEmployee.prototype, "data_flag", void 0);
        SysEmployee = __decorate([
            typeorm.Entity()
        ], SysEmployee);
        return SysEmployee;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysLang = /** @class */ (function () {
        function SysLang() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysLang.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysLang.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysLang.prototype, "name", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysLang.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysLang.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysLang.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysLang.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysLang.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysLang.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysLang.prototype, "data_flag", void 0);
        SysLang = __decorate([
            typeorm.Entity()
        ], SysLang);
        return SysLang;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysMaster = /** @class */ (function () {
        function SysMaster() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "class_code", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('text', { nullable: true }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "remark", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysMaster.prototype, "data_flag", void 0);
        SysMaster = __decorate([
            typeorm.Entity()
        ], SysMaster);
        return SysMaster;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysMasterDetail = /** @class */ (function () {
        function SysMasterDetail() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "class_code", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "master_code", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "code", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('text', { nullable: true }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "remark", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysMasterDetail.prototype, "data_flag", void 0);
        SysMasterDetail = __decorate([
            typeorm.Entity()
        ], SysMasterDetail);
        return SysMasterDetail;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysMessage = /** @class */ (function () {
        function SysMessage() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "code", void 0);
        __decorate([
            typeorm.Column('text'),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "message", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysMessage.prototype, "data_flag", void 0);
        SysMessage = __decorate([
            typeorm.Entity()
        ], SysMessage);
        return SysMessage;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysModule = /** @class */ (function () {
        function SysModule() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "mid", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysModule.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 250 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "icon", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "slug", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "pid", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysModule.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysModule.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysModule.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysModule.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysModule.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysModule.prototype, "data_flag", void 0);
        SysModule = __decorate([
            typeorm.Entity()
        ], SysModule);
        return SysModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SysPage = /** @class */ (function () {
        function SysPage() {
        }
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "company", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "lang", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "mid", void 0);
        __decorate([
            typeorm.PrimaryColumn('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "pid", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 100 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "name", void 0);
        __decorate([
            typeorm.Column('boolean'),
            __metadata("design:type", Object)
        ], SysPage.prototype, "active_flag", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 1000 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "description", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 250 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "icon", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "slug", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysPage.prototype, "sort_no", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 50 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "mapping_table_name", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 50 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "mapping_entity_name", void 0);
        __decorate([
            typeorm.Column('integer'),
            __metadata("design:type", Object)
        ], SysPage.prototype, "change_count", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "create_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysPage.prototype, "create_datetime", void 0);
        __decorate([
            typeorm.Column('character varying', { length: 20 }),
            __metadata("design:type", Object)
        ], SysPage.prototype, "change_emp_id", void 0);
        __decorate([
            typeorm.Column('timestamp without time zone'),
            __metadata("design:type", Object)
        ], SysPage.prototype, "change_datetime", void 0);
        __decorate([
            typeorm.Column('character'),
            __metadata("design:type", Object)
        ], SysPage.prototype, "data_flag", void 0);
        SysPage = __decorate([
            typeorm.Entity()
        ], SysPage);
        return SysPage;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var entities = [
        SysCaption,
        SysCompany,
        SysDefineClass,
        SysDefineDisplay,
        SysDefineGrid,
        SysDefineGroup,
        SysDefineInput,
        SysDefineInputGrid,
        SysDefineMapping,
        SysDefineMappingControl,
        SysDefineMappingFunction,
        SysDefineSearch,
        SysDepartment,
        SysEmployee,
        SysLang,
        SysMaster,
        SysMasterDetail,
        SysMessage,
        SysModule,
        SysPage
    ];

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CaptionService = /** @class */ (function () {
        function CaptionService(captionRepository) {
            this.captionRepository = captionRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        CaptionService.prototype.getCaptions = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.captionRepository.find({
                                company: context.company,
                                lang: context.lang
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        CaptionService.prototype.getCaption = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.captionRepository.find({
                                company: context.company,
                                lang: context.lang,
                                pid: context.pid,
                                type: context.type,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        CaptionService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysCaption)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], CaptionService);
        return CaptionService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CaptionController = /** @class */ (function () {
        function CaptionController(captionService) {
            this.captionService = captionService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        CaptionController.prototype.getCaptions = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.captionService.getCaptions(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        CaptionController.prototype.getCaption = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.captionService.getCaption(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], CaptionController.prototype, "getCaptions", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], CaptionController.prototype, "getCaption", null);
        CaptionController = __decorate([
            common$1.Controller('caption'),
            __metadata("design:paramtypes", [CaptionService])
        ], CaptionController);
        return CaptionController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CompanyService = /** @class */ (function () {
        function CompanyService(companyRepository) {
            this.companyRepository = companyRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        CompanyService.prototype.getCompanies = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.companyRepository.find({
                                lang: context.lang
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        CompanyService.prototype.getCompany = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.companyRepository.find({
                                lang: context.lang,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        CompanyService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysCompany)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], CompanyService);
        return CompanyService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CompanyController = /** @class */ (function () {
        function CompanyController(companyService) {
            this.companyService = companyService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        CompanyController.prototype.getCompanys = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.companyService.getCompanies(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        CompanyController.prototype.getCompany = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.companyService.getCompany(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], CompanyController.prototype, "getCompanys", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], CompanyController.prototype, "getCompany", null);
        CompanyController = __decorate([
            common$1.Controller('company'),
            __metadata("design:paramtypes", [CompanyService])
        ], CompanyController);
        return CompanyController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineClassService = /** @class */ (function () {
        function DefineClassService(defineClassRepository) {
            this.defineClassRepository = defineClassRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineClassService.prototype.getDefineClasses = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineClassRepository.find({
                                company: context.company,
                                lang: context.lang
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        DefineClassService.prototype.getDefineClass = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineClassRepository.find({
                                company: context.company,
                                lang: context.lang,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineClassService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineClass)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineClassService);
        return DefineClassService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineClassController = /** @class */ (function () {
        function DefineClassController(defineClassService) {
            this.defineClassService = defineClassService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineClassController.prototype.getDefineClasses = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineClassService.getDefineClasses(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        DefineClassController.prototype.getDefineClass = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineClassService.getDefineClass(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineClassController.prototype, "getDefineClasses", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineClassController.prototype, "getDefineClass", null);
        DefineClassController = __decorate([
            common$1.Controller('define-class'),
            __metadata("design:paramtypes", [DefineClassService])
        ], DefineClassController);
        return DefineClassController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineDisplayService = /** @class */ (function () {
        function DefineDisplayService(defineDisplayRepository) {
            this.defineDisplayRepository = defineDisplayRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineDisplayService.prototype.getDefineDisplays = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineDisplayRepository.find({
                                company: context.company,
                                emp_id: context.emp_id,
                                pid: context.pid,
                                group_id: context.group_id
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        DefineDisplayService.prototype.getDefineDisplay = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineDisplayRepository.find({
                                company: context.company,
                                emp_id: context.emp_id,
                                pid: context.pid,
                                group_id: context.group_id,
                                item_no: context.item_no
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineDisplayService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineDisplay)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineDisplayService);
        return DefineDisplayService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineDisplayController = /** @class */ (function () {
        function DefineDisplayController(defineDisplayService) {
            this.defineDisplayService = defineDisplayService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineDisplayController.prototype.getDefineDisplays = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineDisplayService.getDefineDisplays(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        DefineDisplayController.prototype.getDefineDisplay = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineDisplayService.getDefineDisplay(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineDisplayController.prototype, "getDefineDisplays", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineDisplayController.prototype, "getDefineDisplay", null);
        DefineDisplayController = __decorate([
            common$1.Controller('define-display'),
            __metadata("design:paramtypes", [DefineDisplayService])
        ], DefineDisplayController);
        return DefineDisplayController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineGridService = /** @class */ (function () {
        function DefineGridService(defineGridRepository) {
            this.defineGridRepository = defineGridRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineGridService.prototype.getDefineGrid = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineGridRepository.find({
                                company: context.company,
                                pid: context.pid,
                                grid_id: context.grid_id
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineGridService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineGrid)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineGridService);
        return DefineGridService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineGridController = /** @class */ (function () {
        function DefineGridController(defineGridService) {
            this.defineGridService = defineGridService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineGridController.prototype.getDefineGrid = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineGridService.getDefineGrid(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineGridController.prototype, "getDefineGrid", null);
        DefineGridController = __decorate([
            common$1.Controller('define-grid'),
            __metadata("design:paramtypes", [DefineGridService])
        ], DefineGridController);
        return DefineGridController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineGroupService = /** @class */ (function () {
        function DefineGroupService(defineGroupRepository) {
            this.defineGroupRepository = defineGroupRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineGroupService.prototype.getDefineGroups = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineGroupRepository.find({
                                company: context.company,
                                pid: context.pid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        DefineGroupService.prototype.getDefineGroup = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineGroupRepository.find({
                                company: context.company,
                                pid: context.pid,
                                group_id: context.group_id
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineGroupService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineGroup)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineGroupService);
        return DefineGroupService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineGroupController = /** @class */ (function () {
        function DefineGroupController(defineGroupService) {
            this.defineGroupService = defineGroupService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineGroupController.prototype.getDefineGroups = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineGroupService.getDefineGroups(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        DefineGroupController.prototype.getDefineGroup = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineGroupService.getDefineGroup(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineGroupController.prototype, "getDefineGroups", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineGroupController.prototype, "getDefineGroup", null);
        DefineGroupController = __decorate([
            common$1.Controller('define-group'),
            __metadata("design:paramtypes", [DefineGroupService])
        ], DefineGroupController);
        return DefineGroupController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineInputService = /** @class */ (function () {
        function DefineInputService(defineInputRepository) {
            this.defineInputRepository = defineInputRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineInputService.prototype.getDefineInput = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineInputRepository.find({
                                company: context.company,
                                pid: context.pid,
                                input_id: context.input_id
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineInputService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineInput)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineInputService);
        return DefineInputService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineInputController = /** @class */ (function () {
        function DefineInputController(defineInputService) {
            this.defineInputService = defineInputService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineInputController.prototype.getDefineInput = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineInputService.getDefineInput(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineInputController.prototype, "getDefineInput", null);
        DefineInputController = __decorate([
            common$1.Controller('define-input'),
            __metadata("design:paramtypes", [DefineInputService])
        ], DefineInputController);
        return DefineInputController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineMappingControlService = /** @class */ (function () {
        function DefineMappingControlService(defineMappingControlRepository) {
            this.defineMappingControlRepository = defineMappingControlRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineMappingControlService.prototype.getDefineMappingControls = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineMappingControlRepository.find({
                                company: context.company,
                                pid: context.pid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        DefineMappingControlService.prototype.getDefineMappingControl = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineMappingControlRepository.find({
                                company: context.company,
                                pid: context.pid,
                                item_no: context.item_no
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineMappingControlService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineMappingControl)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineMappingControlService);
        return DefineMappingControlService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineMappingControlController = /** @class */ (function () {
        function DefineMappingControlController(defineMappingControlService) {
            this.defineMappingControlService = defineMappingControlService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineMappingControlController.prototype.getDefineMappingControls = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineMappingControlService.getDefineMappingControls(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        DefineMappingControlController.prototype.getDefineMappingControl = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineMappingControlService.getDefineMappingControl(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineMappingControlController.prototype, "getDefineMappingControls", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineMappingControlController.prototype, "getDefineMappingControl", null);
        DefineMappingControlController = __decorate([
            common$1.Controller('define-mapping-control'),
            __metadata("design:paramtypes", [DefineMappingControlService])
        ], DefineMappingControlController);
        return DefineMappingControlController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineMappingFunctionService = /** @class */ (function () {
        function DefineMappingFunctionService(defineMappingFunctionRepository) {
            this.defineMappingFunctionRepository = defineMappingFunctionRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineMappingFunctionService.prototype.getDefineMappingFunction = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineMappingFunctionRepository.find({
                                company: context.company,
                                pid: context.pid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineMappingFunctionService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineMappingFunction)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineMappingFunctionService);
        return DefineMappingFunctionService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineMappingFunctionController = /** @class */ (function () {
        function DefineMappingFunctionController(defineMappingFunctionService) {
            this.defineMappingFunctionService = defineMappingFunctionService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineMappingFunctionController.prototype.getDefineMappingFunction = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineMappingFunctionService.getDefineMappingFunction(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineMappingFunctionController.prototype, "getDefineMappingFunction", null);
        DefineMappingFunctionController = __decorate([
            common$1.Controller('define-mapping-function'),
            __metadata("design:paramtypes", [DefineMappingFunctionService])
        ], DefineMappingFunctionController);
        return DefineMappingFunctionController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineMappingService = /** @class */ (function () {
        function DefineMappingService(defineMappingRepository) {
            this.defineMappingRepository = defineMappingRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineMappingService.prototype.getDefineMapping = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineMappingRepository.find({
                                company: context.company,
                                pid: context.pid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineMappingService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineMapping)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineMappingService);
        return DefineMappingService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineMappingController = /** @class */ (function () {
        function DefineMappingController(defineMappingService) {
            this.defineMappingService = defineMappingService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineMappingController.prototype.getDefineMapping = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineMappingService.getDefineMapping(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineMappingController.prototype, "getDefineMapping", null);
        DefineMappingController = __decorate([
            common$1.Controller('define-mapping'),
            __metadata("design:paramtypes", [DefineMappingService])
        ], DefineMappingController);
        return DefineMappingController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineSearchService = /** @class */ (function () {
        function DefineSearchService(defineSearchRepository) {
            this.defineSearchRepository = defineSearchRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineSearchService.prototype.getDefineSearch = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineSearchRepository.find({
                                company: context.company,
                                pid: context.pid,
                                search_id: context.search_id
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineSearchService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineSearch)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineSearchService);
        return DefineSearchService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineSearchController = /** @class */ (function () {
        function DefineSearchController(defineSearchService) {
            this.defineSearchService = defineSearchService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineSearchController.prototype.getDefineSearch = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineSearchService.getDefineSearch(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineSearchController.prototype, "getDefineSearch", null);
        DefineSearchController = __decorate([
            common$1.Controller('define-search'),
            __metadata("design:paramtypes", [DefineSearchService])
        ], DefineSearchController);
        return DefineSearchController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DepartmentService = /** @class */ (function () {
        function DepartmentService(departmentRepository) {
            this.departmentRepository = departmentRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DepartmentService.prototype.getDepartments = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.departmentRepository.find({
                                company: context.company,
                                lang: context.lang
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        DepartmentService.prototype.getDepartment = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.departmentRepository.find({
                                company: context.company,
                                lang: context.lang,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DepartmentService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDepartment)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DepartmentService);
        return DepartmentService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DepartmentController = /** @class */ (function () {
        function DepartmentController(departmentService) {
            this.departmentService = departmentService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DepartmentController.prototype.getDepartments = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.departmentService.getDepartments(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        DepartmentController.prototype.getDepartment = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.departmentService.getDepartment(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DepartmentController.prototype, "getDepartments", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DepartmentController.prototype, "getDepartment", null);
        DepartmentController = __decorate([
            common$1.Controller('department'),
            __metadata("design:paramtypes", [DepartmentService])
        ], DepartmentController);
        return DepartmentController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var EmployeeService = /** @class */ (function () {
        function EmployeeService(employeeRepository) {
            this.employeeRepository = employeeRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        EmployeeService.prototype.getEmployees = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                var params, list, result, index, index, index, index, index, index, index, index, index, index;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            params = new Object();
                            list = [];
                            if (context.name !== undefined) {
                                params['name'] = context.name;
                            }
                            if (context.activeFlag !== undefined) {
                                params['active_flag'] = context.activeFlag;
                            }
                            if (context.address1 !== undefined) {
                                params['address1'] = context.address1;
                            }
                            if (context.address2 !== undefined) {
                                params['address2'] = context.address2;
                            }
                            if (context.tel1 !== undefined) {
                                params['tel1'] = context.tel1;
                            }
                            if (context.tel2 !== undefined) {
                                params['tel2'] = context.tel2;
                            }
                            if (context.email !== undefined) {
                                params['email'] = context.email;
                            }
                            if (context.sortNo !== undefined) {
                                params['sort_no'] = context.sortNo;
                            }
                            if (context.changeCount !== undefined) {
                                params['change_count'] = context.changeCount;
                            }
                            if (context.dataFlag !== undefined) {
                                params['data_flag'] = context.dataFlag;
                            }
                            return [4 /*yield*/, this.employeeRepository.find(__assign({ company: context.company, lang: context.lang }, params))];
                        case 1:
                            result = _a.sent();
                            list = __spread(result);
                            // NOTE Emp code
                            if (context.codeFrom !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.code === context.codeFrom;
                                }));
                                if (index >= 0) {
                                    list = list.slice(index);
                                }
                            }
                            if (context.codeTo !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.code === context.codeTo;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            // NOTE create_emp_id
                            if (context.createEmpFrom !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.create_emp_id === context.createEmpFrom;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            if (context.createEmpTo !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.create_emp_id === context.createEmpTo;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            // NOTE create_datetime
                            if (context.createDatetimeFrom !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.create_datetime === new Date(context.createDatetimeFrom);
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            if (context.createDatetimeTo !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.create_datetime === context.createDatetimeTo;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            // NOTE change_emp_id
                            if (context.changeEmpFrom !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.change_emp_id === context.changeEmpFrom;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            if (context.changeEmpTo !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.change_emp_id === context.changeEmpTo;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            // NOTE change_datetime
                            if (context.changeDatetimeFrom !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.change_datetime === context.changeDatetimeFrom;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            if (context.changeDatetimeTo !== undefined) {
                                index = list.findIndex((/**
                                 * @param {?} element
                                 * @return {?}
                                 */
                                function (element) {
                                    return element.change_datetime === context.changeDatetimeTo;
                                }));
                                if (index >= 0) {
                                    list = list.slice(0, index + 1);
                                }
                            }
                            return [2 /*return*/, list];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        EmployeeService.prototype.getEmployee = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.employeeRepository.find({
                                company: context.company,
                                lang: context.lang,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        EmployeeService.prototype.addEmployee = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                var result, contextVi, contextEn, contextJa;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            if (!(context.code !== '')) return [3 /*break*/, 6];
                            return [4 /*yield*/, this.employeeRepository.find({
                                    company: context.company,
                                    code: context.code
                                })];
                        case 1:
                            result = _a.sent();
                            if (!(result.length <= 0)) return [3 /*break*/, 5];
                            contextVi = __assign({}, context);
                            contextVi.lang = 'vi-VN';
                            contextEn = __assign({}, context);
                            contextEn.lang = 'en-US';
                            contextJa = __assign({}, context);
                            contextJa.lang = 'ja-JP';
                            return [4 /*yield*/, this.employeeRepository.save(contextVi)];
                        case 2:
                            _a.sent();
                            return [4 /*yield*/, this.employeeRepository.save(contextEn)];
                        case 3:
                            _a.sent();
                            return [4 /*yield*/, this.employeeRepository.save(contextJa)];
                        case 4:
                            _a.sent();
                            return [2 /*return*/, true];
                        case 5: return [2 /*return*/, false];
                        case 6: return [2 /*return*/, false];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        EmployeeService.prototype.updateEmployee = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                var result, contextVi, contextEn, contextJa;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            if (!(context.code !== '')) return [3 /*break*/, 6];
                            return [4 /*yield*/, this.employeeRepository.find({
                                    company: context.company,
                                    code: context.code
                                })];
                        case 1:
                            result = _a.sent();
                            if (!(result.length > 0)) return [3 /*break*/, 5];
                            context.change_count++;
                            context.change_datetime = null;
                            contextVi = __assign({}, context);
                            contextVi.lang = 'vi-VN';
                            contextEn = __assign({}, context);
                            contextEn.lang = 'en-US';
                            contextJa = __assign({}, context);
                            contextJa.lang = 'ja-JP';
                            return [4 /*yield*/, this.employeeRepository.update([contextVi.company, contextVi.lang, contextVi.code], contextVi)];
                        case 2:
                            _a.sent();
                            return [4 /*yield*/, this.employeeRepository.update([contextEn.company, contextEn.lang, contextEn.code], contextEn)];
                        case 3:
                            _a.sent();
                            return [4 /*yield*/, this.employeeRepository.update([contextJa.company, contextJa.lang, contextJa.code], contextJa)];
                        case 4:
                            _a.sent();
                            return [2 /*return*/, true];
                        case 5: return [2 /*return*/, false];
                        case 6: return [2 /*return*/, false];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        EmployeeService.prototype.deleteEmployee = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                var result;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.employeeRepository.delete({ code: context })];
                        case 1:
                            result = _a.sent();
                            if (result.affected > 0) {
                                return [2 /*return*/, true];
                            }
                            return [2 /*return*/, false];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        EmployeeService.prototype.deleteEmployees = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                var result;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.employeeRepository.delete(context)];
                        case 1:
                            result = _a.sent();
                            if (result.affected > 0) {
                                return [2 /*return*/, true];
                            }
                            return [2 /*return*/, false];
                    }
                });
            });
        };
        EmployeeService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysEmployee)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], EmployeeService);
        return EmployeeService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var EmployeeController = /** @class */ (function () {
        function EmployeeController(employeeService) {
            this.employeeService = employeeService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        EmployeeController.prototype.getEmployees = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.employeeService.getEmployees(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        EmployeeController.prototype.getEmployee = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.employeeService.getEmployee(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        EmployeeController.prototype.addEmployee = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.employeeService.addEmployee(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        EmployeeController.prototype.updateEmployee = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.employeeService.updateEmployee(body);
        };
        /**
         * @param {?} params
         * @return {?}
         */
        EmployeeController.prototype.deleteEmployee = /**
         * @param {?} params
         * @return {?}
         */
        function (params) {
            return this.employeeService.deleteEmployee(params.id);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        EmployeeController.prototype.deleteEmployees = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.employeeService.deleteEmployees(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], EmployeeController.prototype, "getEmployees", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], EmployeeController.prototype, "getEmployee", null);
        __decorate([
            common$1.Post('add'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], EmployeeController.prototype, "addEmployee", null);
        __decorate([
            common$1.Post('update'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], EmployeeController.prototype, "updateEmployee", null);
        __decorate([
            common$1.Delete(':id'),
            __param(0, common$1.Param()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], EmployeeController.prototype, "deleteEmployee", null);
        __decorate([
            common$1.Post('delete/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], EmployeeController.prototype, "deleteEmployees", null);
        EmployeeController = __decorate([
            common$1.Controller('employee'),
            __metadata("design:paramtypes", [EmployeeService])
        ], EmployeeController);
        return EmployeeController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var LangService = /** @class */ (function () {
        function LangService(langRepository) {
            this.langRepository = langRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        LangService.prototype.getLangs = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    return [2 /*return*/, this.langRepository.find({
                            company: context.company
                        })];
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        LangService.prototype.getLang = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    return [2 /*return*/, this.langRepository.find({
                            company: context.company,
                            code: context.code
                        })];
                });
            });
        };
        LangService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysLang)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], LangService);
        return LangService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var LangController = /** @class */ (function () {
        function LangController(langService) {
            this.langService = langService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        LangController.prototype.getLangs = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.langService.getLangs(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        LangController.prototype.getLang = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.langService.getLang(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], LangController.prototype, "getLangs", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], LangController.prototype, "getLang", null);
        LangController = __decorate([
            common$1.Controller('lang'),
            __metadata("design:paramtypes", [LangService])
        ], LangController);
        return LangController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MasterDetailService = /** @class */ (function () {
        function MasterDetailService(masterDetailRepository) {
            this.masterDetailRepository = masterDetailRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        MasterDetailService.prototype.getMasterDetails = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.masterDetailRepository.find({
                                company: context.company,
                                lang: context.lang,
                                class_code: context.class_code,
                                master_code: context.master_code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        MasterDetailService.prototype.getMasterDetail = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.masterDetailRepository.find({
                                company: context.company,
                                lang: context.lang,
                                class_code: context.class_code,
                                master_code: context.master_code,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        MasterDetailService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysMasterDetail)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], MasterDetailService);
        return MasterDetailService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MasterDetailController = /** @class */ (function () {
        function MasterDetailController(masterDetailService) {
            this.masterDetailService = masterDetailService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        MasterDetailController.prototype.getMasterDetails = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.masterDetailService.getMasterDetails(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        MasterDetailController.prototype.getMasterDetail = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.masterDetailService.getMasterDetail(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], MasterDetailController.prototype, "getMasterDetails", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], MasterDetailController.prototype, "getMasterDetail", null);
        MasterDetailController = __decorate([
            common$1.Controller('master-detail'),
            __metadata("design:paramtypes", [MasterDetailService])
        ], MasterDetailController);
        return MasterDetailController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MasterService = /** @class */ (function () {
        function MasterService(masterRepository) {
            this.masterRepository = masterRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        MasterService.prototype.getMasters = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.masterRepository.find({
                                company: context.company,
                                lang: context.lang,
                                class_code: context.class_code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        MasterService.prototype.getMaster = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.masterRepository.find({
                                company: context.company,
                                lang: context.lang,
                                class_code: context.class_code,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        MasterService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysMaster)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], MasterService);
        return MasterService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MasterController = /** @class */ (function () {
        function MasterController(masterService) {
            this.masterService = masterService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        MasterController.prototype.getMasters = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.masterService.getMasters(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        MasterController.prototype.getMaster = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.masterService.getMaster(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], MasterController.prototype, "getMasters", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], MasterController.prototype, "getMaster", null);
        MasterController = __decorate([
            common$1.Controller('master'),
            __metadata("design:paramtypes", [MasterService])
        ], MasterController);
        return MasterController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MessageService = /** @class */ (function () {
        function MessageService(messageRepository) {
            this.messageRepository = messageRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        MessageService.prototype.getMessage = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.messageRepository.find({
                                company: context.company,
                                lang: context.lang,
                                code: context.code
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        MessageService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysMessage)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], MessageService);
        return MessageService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MessageController = /** @class */ (function () {
        function MessageController(messageService) {
            this.messageService = messageService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        MessageController.prototype.getMessage = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.messageService.getMessage(body);
        };
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], MessageController.prototype, "getMessage", null);
        MessageController = __decorate([
            common$1.Controller('message'),
            __metadata("design:paramtypes", [MessageService])
        ], MessageController);
        return MessageController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ModuleService = /** @class */ (function () {
        function ModuleService(moduleRepository) {
            this.moduleRepository = moduleRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        ModuleService.prototype.getModules = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.moduleRepository.find({
                                company: context.company,
                                lang: context.lang
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        ModuleService.prototype.getModule = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.moduleRepository.find({
                                company: context.company,
                                lang: context.lang,
                                mid: context.mid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        ModuleService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysModule)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], ModuleService);
        return ModuleService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ModuleController = /** @class */ (function () {
        function ModuleController(moduleService) {
            this.moduleService = moduleService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        ModuleController.prototype.getModules = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.moduleService.getModules(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        ModuleController.prototype.getModule = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.moduleService.getModule(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], ModuleController.prototype, "getModules", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], ModuleController.prototype, "getModule", null);
        ModuleController = __decorate([
            common$1.Controller('module'),
            __metadata("design:paramtypes", [ModuleService])
        ], ModuleController);
        return ModuleController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PageService = /** @class */ (function () {
        function PageService(pageRepository) {
            this.pageRepository = pageRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        PageService.prototype.getPages = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.pageRepository.find({
                                company: context.company,
                                lang: context.lang,
                                mid: context.mid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @param {?} context
         * @return {?}
         */
        PageService.prototype.getPage = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.pageRepository.find({
                                company: context.company,
                                lang: context.lang,
                                mid: context.mid,
                                pid: context.pid
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        PageService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysPage)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], PageService);
        return PageService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PageController = /** @class */ (function () {
        function PageController(pageService) {
            this.pageService = pageService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        PageController.prototype.getPages = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.pageService.getPages(body);
        };
        /**
         * @param {?} body
         * @return {?}
         */
        PageController.prototype.getPage = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.pageService.getPage(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], PageController.prototype, "getPages", null);
        __decorate([
            common$1.Post('get/one'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], PageController.prototype, "getPage", null);
        PageController = __decorate([
            common$1.Controller('page'),
            __metadata("design:paramtypes", [PageService])
        ], PageController);
        return PageController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineInputGridService = /** @class */ (function () {
        function DefineInputGridService(defineInputGridRepository) {
            this.defineInputGridRepository = defineInputGridRepository;
        }
        /**
         * @param {?} context
         * @return {?}
         */
        DefineInputGridService.prototype.getDefineInputGrid = /**
         * @param {?} context
         * @return {?}
         */
        function (context) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.defineInputGridRepository.find({
                                company: context.company,
                                pid: context.pid,
                                input_grid_id: context.input_grid_id
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        DefineInputGridService = __decorate([
            common$1.Injectable(),
            __param(0, typeorm$1.InjectRepository(SysDefineInputGrid)),
            __metadata("design:paramtypes", [typeorm.Repository])
        ], DefineInputGridService);
        return DefineInputGridService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DefineInputGridController = /** @class */ (function () {
        function DefineInputGridController(defineInputGridService) {
            this.defineInputGridService = defineInputGridService;
        }
        /**
         * @param {?} body
         * @return {?}
         */
        DefineInputGridController.prototype.getDefineInputGrid = /**
         * @param {?} body
         * @return {?}
         */
        function (body) {
            return this.defineInputGridService.getDefineInputGrid(body);
        };
        __decorate([
            common$1.Post('get/many'),
            __param(0, common$1.Body()),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Object]),
            __metadata("design:returntype", void 0)
        ], DefineInputGridController.prototype, "getDefineInputGrid", null);
        DefineInputGridController = __decorate([
            common$1.Controller('define-input-grid'),
            __metadata("design:paramtypes", [DefineInputGridService])
        ], DefineInputGridController);
        return DefineInputGridController;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var controllers = [
        CaptionController,
        CompanyController,
        DefineClassController,
        DefineDisplayController,
        DefineGridController,
        DefineGroupController,
        DefineInputGridController,
        DefineInputController,
        DefineMappingControlController,
        DefineMappingFunctionController,
        DefineMappingController,
        DefineSearchController,
        DepartmentController,
        EmployeeController,
        LangController,
        MasterDetailController,
        MasterController,
        MessageController,
        ModuleController,
        PageController
    ];

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var services = [
        CaptionService,
        CompanyService,
        DefineClassService,
        DefineDisplayService,
        DefineGridService,
        DefineGroupService,
        DefineInputGridService,
        DefineInputService,
        DefineMappingControlService,
        DefineMappingFunctionService,
        DefineMappingService,
        DefineSearchService,
        DepartmentService,
        EmployeeService,
        LangService,
        MasterDetailService,
        MasterService,
        MessageService,
        ModuleService,
        PageService
    ];

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AitCoreModule = /** @class */ (function () {
        function AitCoreModule() {
        }
        AitCoreModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            router.RouterModule.forChild([
                            /* {path: '', pathMatch: 'full', component: InsertYourComponentHere} */
                            ])
                        ]
                    },] }
        ];
        return AitCoreModule;
    }());

    exports.AitCoreModule = AitCoreModule;
    exports.controllers = controllers;
    exports.dbConfig = dbConfig;
    exports.entities = entities;
    exports.services = services;
    exports.a = SysCaption;
    exports.b = SysCompany;
    exports.ba = DefineDisplayController;
    exports.bb = DefineDisplayService;
    exports.bc = DefineGridController;
    exports.bd = DefineGridService;
    exports.be = DefineGroupController;
    exports.bf = DefineGroupService;
    exports.bg = DefineInputGridController;
    exports.bh = DefineInputGridService;
    exports.bi = DefineInputController;
    exports.bj = DefineInputService;
    exports.bk = DefineMappingControlController;
    exports.bl = DefineMappingControlService;
    exports.bm = DefineMappingFunctionController;
    exports.bn = DefineMappingFunctionService;
    exports.bo = DefineMappingController;
    exports.bp = DefineMappingService;
    exports.bq = DefineSearchController;
    exports.br = DefineSearchService;
    exports.bs = DepartmentController;
    exports.bt = DepartmentService;
    exports.bu = EmployeeController;
    exports.bv = EmployeeService;
    exports.bw = LangController;
    exports.bx = LangService;
    exports.by = MasterDetailController;
    exports.bz = MasterDetailService;
    exports.c = SysDefineClass;
    exports.ca = MasterController;
    exports.cb = MasterService;
    exports.cc = MessageController;
    exports.cd = MessageService;
    exports.ce = ModuleController;
    exports.cf = ModuleService;
    exports.cg = PageController;
    exports.ch = PageService;
    exports.d = SysDefineDisplay;
    exports.e = SysDefineGrid;
    exports.f = SysDefineGroup;
    exports.g = SysDefineInput;
    exports.h = SysDefineInputGrid;
    exports.i = SysDefineMapping;
    exports.j = SysDefineMappingControl;
    exports.k = SysDefineMappingFunction;
    exports.l = SysDefineSearch;
    exports.m = SysDepartment;
    exports.n = SysEmployee;
    exports.o = SysLang;
    exports.p = SysMaster;
    exports.q = SysMasterDetail;
    exports.r = SysMessage;
    exports.s = SysModule;
    exports.t = SysPage;
    exports.u = CaptionController;
    exports.v = CaptionService;
    exports.w = CompanyController;
    exports.x = CompanyService;
    exports.y = DefineClassController;
    exports.z = DefineClassService;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
