// @aureole/core/esm5/lib/services/caption.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysCaption } from '../entities/sys_caption.entity';
var CaptionService = /** @class */ (function () {
    function CaptionService(captionRepository) {
        this.captionRepository = captionRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    CaptionService.prototype.getCaptions = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.captionRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    CaptionService.prototype.getCaption = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.captionRepository.find({
                            company: context.company,
                            lang: context.lang,
                            pid: context.pid,
                            type: context.type,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    CaptionService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysCaption)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], CaptionService);
    return CaptionService;
}());
export { CaptionService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    CaptionService.prototype.captionRepository;
}
