// @aureole/core/esm5/lib/services/company.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysCompany } from '../entities/sys_company.entity';
var CompanyService = /** @class */ (function () {
    function CompanyService(companyRepository) {
        this.companyRepository = companyRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    CompanyService.prototype.getCompanies = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.companyRepository.find({
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    CompanyService.prototype.getCompany = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.companyRepository.find({
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    CompanyService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysCompany)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], CompanyService);
    return CompanyService;
}());
export { CompanyService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    CompanyService.prototype.companyRepository;
}
