// @aureole/core/esm5/lib/services/define-mapping-control.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineMappingControl } from '../entities/sys_define_mapping_control.entity';
var DefineMappingControlService = /** @class */ (function () {
    function DefineMappingControlService(defineMappingControlRepository) {
        this.defineMappingControlRepository = defineMappingControlRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineMappingControlService.prototype.getDefineMappingControls = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineMappingControlRepository.find({
                            company: context.company,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineMappingControlService.prototype.getDefineMappingControl = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineMappingControlRepository.find({
                            company: context.company,
                            pid: context.pid,
                            item_no: context.item_no
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineMappingControlService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineMappingControl)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineMappingControlService);
    return DefineMappingControlService;
}());
export { DefineMappingControlService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineMappingControlService.prototype.defineMappingControlRepository;
}
