// @aureole/core/esm5/lib/services/define-input.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineInput } from '../entities/sys_define_input.entity';
var DefineInputService = /** @class */ (function () {
    function DefineInputService(defineInputRepository) {
        this.defineInputRepository = defineInputRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineInputService.prototype.getDefineInput = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineInputRepository.find({
                            company: context.company,
                            pid: context.pid,
                            input_id: context.input_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineInputService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineInput)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineInputService);
    return DefineInputService;
}());
export { DefineInputService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineInputService.prototype.defineInputRepository;
}
