// @aureole/core/esm5/lib/services/define-class.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { SysDefineClass } from '../entities/sys_define_class.entity';
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
var DefineClassService = /** @class */ (function () {
    function DefineClassService(defineClassRepository) {
        this.defineClassRepository = defineClassRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineClassService.prototype.getDefineClasses = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineClassRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineClassService.prototype.getDefineClass = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineClassRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineClassService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineClass)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineClassService);
    return DefineClassService;
}());
export { DefineClassService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineClassService.prototype.defineClassRepository;
}
