// @aureole/core/esm5/lib/services/define-display.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineDisplay } from '../entities/sys_define_display.entity';
var DefineDisplayService = /** @class */ (function () {
    function DefineDisplayService(defineDisplayRepository) {
        this.defineDisplayRepository = defineDisplayRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineDisplayService.prototype.getDefineDisplays = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineDisplayRepository.find({
                            company: context.company,
                            emp_id: context.emp_id,
                            pid: context.pid,
                            group_id: context.group_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineDisplayService.prototype.getDefineDisplay = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineDisplayRepository.find({
                            company: context.company,
                            emp_id: context.emp_id,
                            pid: context.pid,
                            group_id: context.group_id,
                            item_no: context.item_no
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineDisplayService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineDisplay)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineDisplayService);
    return DefineDisplayService;
}());
export { DefineDisplayService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineDisplayService.prototype.defineDisplayRepository;
}
