// @aureole/core/esm5/lib/services/define-input-grid.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineInputGrid } from '../entities/sys_define_input_grid.entity';
var DefineInputGridService = /** @class */ (function () {
    function DefineInputGridService(defineInputGridRepository) {
        this.defineInputGridRepository = defineInputGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineInputGridService.prototype.getDefineInputGrid = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineInputGridRepository.find({
                            company: context.company,
                            pid: context.pid,
                            input_grid_id: context.input_grid_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineInputGridService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineInputGrid)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineInputGridService);
    return DefineInputGridService;
}());
export { DefineInputGridService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineInputGridService.prototype.defineInputGridRepository;
}
