// @aureole/core/esm5/lib/services/define-grid.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineGrid } from '../entities/sys_define_grid.entity';
var DefineGridService = /** @class */ (function () {
    function DefineGridService(defineGridRepository) {
        this.defineGridRepository = defineGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineGridService.prototype.getDefineGrid = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineGridRepository.find({
                            company: context.company,
                            pid: context.pid,
                            grid_id: context.grid_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineGridService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineGrid)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineGridService);
    return DefineGridService;
}());
export { DefineGridService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineGridService.prototype.defineGridRepository;
}
