// @aureole/core/esm5/lib/services/define-search.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineSearch } from '../entities/sys_define_search.entity';
var DefineSearchService = /** @class */ (function () {
    function DefineSearchService(defineSearchRepository) {
        this.defineSearchRepository = defineSearchRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineSearchService.prototype.getDefineSearch = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineSearchRepository.find({
                            company: context.company,
                            pid: context.pid,
                            search_id: context.search_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineSearchService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineSearch)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineSearchService);
    return DefineSearchService;
}());
export { DefineSearchService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineSearchService.prototype.defineSearchRepository;
}
