// @aureole/core/esm5/lib/services/define-group.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDefineGroup } from '../entities/sys_define_group.entity';
var DefineGroupService = /** @class */ (function () {
    function DefineGroupService(defineGroupRepository) {
        this.defineGroupRepository = defineGroupRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineGroupService.prototype.getDefineGroups = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineGroupRepository.find({
                            company: context.company,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineGroupService.prototype.getDefineGroup = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineGroupRepository.find({
                            company: context.company,
                            pid: context.pid,
                            group_id: context.group_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineGroupService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDefineGroup)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DefineGroupService);
    return DefineGroupService;
}());
export { DefineGroupService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DefineGroupService.prototype.defineGroupRepository;
}
