// @aureole/core/esm5/lib/services/department.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysDepartment } from '../entities/sys_department.entity';
var DepartmentService = /** @class */ (function () {
    function DepartmentService(departmentRepository) {
        this.departmentRepository = departmentRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DepartmentService.prototype.getDepartments = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.departmentRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DepartmentService.prototype.getDepartment = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.departmentRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DepartmentService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysDepartment)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], DepartmentService);
    return DepartmentService;
}());
export { DepartmentService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DepartmentService.prototype.departmentRepository;
}
