// @aureole/core/esm5/lib/services/employee.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysEmployee } from '../entities/sys_employee.entity';
var EmployeeService = /** @class */ (function () {
    function EmployeeService(employeeRepository) {
        this.employeeRepository = employeeRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.getEmployees = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var params, list, result, index, index, index, index, index, index, index, index, index, index;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        params = new Object();
                        list = [];
                        if (context.name !== undefined) {
                            params['name'] = context.name;
                        }
                        if (context.activeFlag !== undefined) {
                            params['active_flag'] = context.activeFlag;
                        }
                        if (context.address1 !== undefined) {
                            params['address1'] = context.address1;
                        }
                        if (context.address2 !== undefined) {
                            params['address2'] = context.address2;
                        }
                        if (context.tel1 !== undefined) {
                            params['tel1'] = context.tel1;
                        }
                        if (context.tel2 !== undefined) {
                            params['tel2'] = context.tel2;
                        }
                        if (context.email !== undefined) {
                            params['email'] = context.email;
                        }
                        if (context.sortNo !== undefined) {
                            params['sort_no'] = context.sortNo;
                        }
                        if (context.changeCount !== undefined) {
                            params['change_count'] = context.changeCount;
                        }
                        if (context.dataFlag !== undefined) {
                            params['data_flag'] = context.dataFlag;
                        }
                        return [4 /*yield*/, this.employeeRepository.find(tslib_1.__assign({ company: context.company, lang: context.lang }, params))];
                    case 1:
                        result = _a.sent();
                        list = tslib_1.__spread(result);
                        // NOTE Emp code
                        if (context.codeFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.code === context.codeFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(index);
                            }
                        }
                        if (context.codeTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.code === context.codeTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE create_emp_id
                        if (context.createEmpFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_emp_id === context.createEmpFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.createEmpTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_emp_id === context.createEmpTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE create_datetime
                        if (context.createDatetimeFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_datetime === new Date(context.createDatetimeFrom);
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.createDatetimeTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_datetime === context.createDatetimeTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE change_emp_id
                        if (context.changeEmpFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_emp_id === context.changeEmpFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.changeEmpTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_emp_id === context.changeEmpTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE change_datetime
                        if (context.changeDatetimeFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_datetime === context.changeDatetimeFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.changeDatetimeTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_datetime === context.changeDatetimeTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        return [2 /*return*/, list];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.getEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.employeeRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.addEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result, contextVi, contextEn, contextJa;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!(context.code !== '')) return [3 /*break*/, 6];
                        return [4 /*yield*/, this.employeeRepository.find({
                                company: context.company,
                                code: context.code
                            })];
                    case 1:
                        result = _a.sent();
                        if (!(result.length <= 0)) return [3 /*break*/, 5];
                        contextVi = tslib_1.__assign({}, context);
                        contextVi.lang = 'vi-VN';
                        contextEn = tslib_1.__assign({}, context);
                        contextEn.lang = 'en-US';
                        contextJa = tslib_1.__assign({}, context);
                        contextJa.lang = 'ja-JP';
                        return [4 /*yield*/, this.employeeRepository.save(contextVi)];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.save(contextEn)];
                    case 3:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.save(contextJa)];
                    case 4:
                        _a.sent();
                        return [2 /*return*/, true];
                    case 5: return [2 /*return*/, false];
                    case 6: return [2 /*return*/, false];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.updateEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result, contextVi, contextEn, contextJa;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!(context.code !== '')) return [3 /*break*/, 6];
                        return [4 /*yield*/, this.employeeRepository.find({
                                company: context.company,
                                code: context.code
                            })];
                    case 1:
                        result = _a.sent();
                        if (!(result.length > 0)) return [3 /*break*/, 5];
                        context.change_count++;
                        context.change_datetime = null;
                        contextVi = tslib_1.__assign({}, context);
                        contextVi.lang = 'vi-VN';
                        contextEn = tslib_1.__assign({}, context);
                        contextEn.lang = 'en-US';
                        contextJa = tslib_1.__assign({}, context);
                        contextJa.lang = 'ja-JP';
                        return [4 /*yield*/, this.employeeRepository.update([contextVi.company, contextVi.lang, contextVi.code], contextVi)];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.update([contextEn.company, contextEn.lang, contextEn.code], contextEn)];
                    case 3:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.update([contextJa.company, contextJa.lang, contextJa.code], contextJa)];
                    case 4:
                        _a.sent();
                        return [2 /*return*/, true];
                    case 5: return [2 /*return*/, false];
                    case 6: return [2 /*return*/, false];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.deleteEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.employeeRepository.delete({ code: context })];
                    case 1:
                        result = _a.sent();
                        if (result.affected > 0) {
                            return [2 /*return*/, true];
                        }
                        return [2 /*return*/, false];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.deleteEmployees = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.employeeRepository.delete(context)];
                    case 1:
                        result = _a.sent();
                        if (result.affected > 0) {
                            return [2 /*return*/, true];
                        }
                        return [2 /*return*/, false];
                }
            });
        });
    };
    EmployeeService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysEmployee)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], EmployeeService);
    return EmployeeService;
}());
export { EmployeeService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    EmployeeService.prototype.employeeRepository;
}
