// @aureole/core/esm5/lib/services/master-detail.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysMasterDetail } from '../entities/sys_master_detail.entity';
var MasterDetailService = /** @class */ (function () {
    function MasterDetailService(masterDetailRepository) {
        this.masterDetailRepository = masterDetailRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    MasterDetailService.prototype.getMasterDetails = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterDetailRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code,
                            master_code: context.master_code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    MasterDetailService.prototype.getMasterDetail = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterDetailRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code,
                            master_code: context.master_code,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    MasterDetailService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysMasterDetail)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], MasterDetailService);
    return MasterDetailService;
}());
export { MasterDetailService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MasterDetailService.prototype.masterDetailRepository;
}
