// @aureole/core/esm5/lib/services/master.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysMaster } from '../entities/sys_master.entity';
var MasterService = /** @class */ (function () {
    function MasterService(masterRepository) {
        this.masterRepository = masterRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    MasterService.prototype.getMasters = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    MasterService.prototype.getMaster = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    MasterService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysMaster)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], MasterService);
    return MasterService;
}());
export { MasterService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MasterService.prototype.masterRepository;
}
