// @aureole/core/esm5/lib/services/module.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysModule } from '../entities/sys_module.entity';
var ModuleService = /** @class */ (function () {
    function ModuleService(moduleRepository) {
        this.moduleRepository = moduleRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    ModuleService.prototype.getModules = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.moduleRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    ModuleService.prototype.getModule = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.moduleRepository.find({
                            company: context.company,
                            lang: context.lang,
                            mid: context.mid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    ModuleService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysModule)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], ModuleService);
    return ModuleService;
}());
export { ModuleService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ModuleService.prototype.moduleRepository;
}
