// @aureole/core/esm5/lib/services/page.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { SysPage } from '../entities/sys_page.entity';
var PageService = /** @class */ (function () {
    function PageService(pageRepository) {
        this.pageRepository = pageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    PageService.prototype.getPages = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.pageRepository.find({
                            company: context.company,
                            lang: context.lang,
                            mid: context.mid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    PageService.prototype.getPage = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.pageRepository.find({
                            company: context.company,
                            lang: context.lang,
                            mid: context.mid,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    PageService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__param(0, InjectRepository(SysPage)),
        tslib_1.__metadata("design:paramtypes", [Repository])
    ], PageService);
    return PageService;
}());
export { PageService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    PageService.prototype.pageRepository;
}
