// @aureole/core/fesm5/ait-ait-core.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { __decorate, __metadata, __awaiter, __generator, __param, __assign, __spread } from 'tslib';
import { PrimaryColumn, Column, Entity, Repository } from 'typeorm';
import { Injectable, Post, Body, Controller, Delete, Param } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
var dbConfig = {
    type: 'postgres',
    host: '192.168.136.12',
    port: 1102,
    username: 'ait',
    password: 'ait',
    database: 'ait',
    schema: 'ait_pm_dev',
    synchronize: true
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysCaption = /** @class */ (function () {
    function SysCaption() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "type", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "code", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "name", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysCaption.prototype, "data_flag", void 0);
    SysCaption = __decorate([
        Entity()
    ], SysCaption);
    return SysCaption;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysCompany = /** @class */ (function () {
    function SysCompany() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "active_flag", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "address1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "address2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "tel1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "tel2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "email", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysCompany.prototype, "data_flag", void 0);
    SysCompany = __decorate([
        Entity()
    ], SysCompany);
    return SysCompany;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineClass = /** @class */ (function () {
    function SysDefineClass() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "active_flag", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "remark", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "sort_no", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "table_ref", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineClass.prototype, "data_flag", void 0);
    SysDefineClass = __decorate([
        Entity()
    ], SysDefineClass);
    return SysDefineClass;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineDisplay = /** @class */ (function () {
    function SysDefineDisplay() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "emp_id", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "group_id", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "item_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "disp_no", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "item_disp", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "width", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "height", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "row_start", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "row_end", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "col_start", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "col_end", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "color_bg", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "color_text", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "class", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "stype", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineDisplay.prototype, "data_flag", void 0);
    SysDefineDisplay = __decorate([
        Entity()
    ], SysDefineDisplay);
    return SysDefineDisplay;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineGrid = /** @class */ (function () {
    function SysDefineGrid() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "grid_id", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "item_no", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "data_type", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 100 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "data_format", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "cap_type", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "cap_code", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineGrid.prototype, "data_flag", void 0);
    SysDefineGrid = __decorate([
        Entity()
    ], SysDefineGrid);
    return SysDefineGrid;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineGroup = /** @class */ (function () {
    function SysDefineGroup() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "group_id", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "cap_type", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "cap_code", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineGroup.prototype, "data_flag", void 0);
    SysDefineGroup = __decorate([
        Entity()
    ], SysDefineGroup);
    return SysDefineGroup;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineInput = /** @class */ (function () {
    function SysDefineInput() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "input_id", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "item_no", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "input_type", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "cap_type", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "cap_code", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "default_value", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineInput.prototype, "data_flag", void 0);
    SysDefineInput = __decorate([
        Entity()
    ], SysDefineInput);
    return SysDefineInput;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineInputGrid = /** @class */ (function () {
    function SysDefineInputGrid() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "input_grid_id", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "item_no", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "input_type", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "cap_type", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "cap_code", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "default_value", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineInputGrid.prototype, "data_flag", void 0);
    SysDefineInputGrid = __decorate([
        Entity()
    ], SysDefineInputGrid);
    return SysDefineInputGrid;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineMapping = /** @class */ (function () {
    function SysDefineMapping() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "mapping_group_id", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "mapping_item_no", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 250 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "mapping_item_field", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "mapping_table_name", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "mapping_column_name", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "mapping_data_type", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineMapping.prototype, "data_flag", void 0);
    SysDefineMapping = __decorate([
        Entity()
    ], SysDefineMapping);
    return SysDefineMapping;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineMappingControl = /** @class */ (function () {
    function SysDefineMappingControl() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "item_no", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "control_type", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_table", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column0", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column3", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column4", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column5", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column6", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column7", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column8", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "mapping_column9", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineMappingControl.prototype, "data_flag", void 0);
    SysDefineMappingControl = __decorate([
        Entity()
    ], SysDefineMappingControl);
    return SysDefineMappingControl;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineMappingFunction = /** @class */ (function () {
    function SysDefineMappingFunction() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "screen_type", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "function_type", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "active_flag", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineMappingFunction.prototype, "data_flag", void 0);
    SysDefineMappingFunction = __decorate([
        Entity()
    ], SysDefineMappingFunction);
    return SysDefineMappingFunction;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDefineSearch = /** @class */ (function () {
    function SysDefineSearch() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "pid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "search_id", void 0);
    __decorate([
        PrimaryColumn('integer'),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "item_no", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "search_type", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "cap_type", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "cap_code", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "default_value", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDefineSearch.prototype, "data_flag", void 0);
    SysDefineSearch = __decorate([
        Entity()
    ], SysDefineSearch);
    return SysDefineSearch;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysDepartment = /** @class */ (function () {
    function SysDepartment() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "active_flag", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "address1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "address2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "tel1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "tel2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "email", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysDepartment.prototype, "data_flag", void 0);
    SysDepartment = __decorate([
        Entity()
    ], SysDepartment);
    return SysDepartment;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysEmployee = /** @class */ (function () {
    function SysEmployee() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "active_flag", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "address1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 250 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "address2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "tel1", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "tel2", void 0);
    __decorate([
        Column('character varying', { nullable: true, length: 50 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "email", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysEmployee.prototype, "data_flag", void 0);
    SysEmployee = __decorate([
        Entity()
    ], SysEmployee);
    return SysEmployee;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysLang = /** @class */ (function () {
    function SysLang() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysLang.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysLang.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysLang.prototype, "name", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysLang.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysLang.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysLang.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysLang.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysLang.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysLang.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysLang.prototype, "data_flag", void 0);
    SysLang = __decorate([
        Entity()
    ], SysLang);
    return SysLang;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysMaster = /** @class */ (function () {
    function SysMaster() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "class_code", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "active_flag", void 0);
    __decorate([
        Column('text', { nullable: true }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "remark", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysMaster.prototype, "data_flag", void 0);
    SysMaster = __decorate([
        Entity()
    ], SysMaster);
    return SysMaster;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysMasterDetail = /** @class */ (function () {
    function SysMasterDetail() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "class_code", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "master_code", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "code", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "active_flag", void 0);
    __decorate([
        Column('text', { nullable: true }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "remark", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysMasterDetail.prototype, "data_flag", void 0);
    SysMasterDetail = __decorate([
        Entity()
    ], SysMasterDetail);
    return SysMasterDetail;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysMessage = /** @class */ (function () {
    function SysMessage() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "code", void 0);
    __decorate([
        Column('text'),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "message", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysMessage.prototype, "data_flag", void 0);
    SysMessage = __decorate([
        Entity()
    ], SysMessage);
    return SysMessage;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysModule = /** @class */ (function () {
    function SysModule() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "mid", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysModule.prototype, "active_flag", void 0);
    __decorate([
        Column('character varying', { length: 250 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "icon", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "slug", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "pid", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysModule.prototype, "sort_no", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysModule.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysModule.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysModule.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysModule.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysModule.prototype, "data_flag", void 0);
    SysModule = __decorate([
        Entity()
    ], SysModule);
    return SysModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var SysPage = /** @class */ (function () {
    function SysPage() {
    }
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "company", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "lang", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "mid", void 0);
    __decorate([
        PrimaryColumn('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "pid", void 0);
    __decorate([
        Column('character varying', { length: 100 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "name", void 0);
    __decorate([
        Column('boolean'),
        __metadata("design:type", Object)
    ], SysPage.prototype, "active_flag", void 0);
    __decorate([
        Column('character varying', { length: 1000 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "description", void 0);
    __decorate([
        Column('character varying', { length: 250 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "icon", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "slug", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysPage.prototype, "sort_no", void 0);
    __decorate([
        Column('character varying', { length: 50 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "mapping_table_name", void 0);
    __decorate([
        Column('character varying', { length: 50 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "mapping_entity_name", void 0);
    __decorate([
        Column('integer'),
        __metadata("design:type", Object)
    ], SysPage.prototype, "change_count", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "create_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysPage.prototype, "create_datetime", void 0);
    __decorate([
        Column('character varying', { length: 20 }),
        __metadata("design:type", Object)
    ], SysPage.prototype, "change_emp_id", void 0);
    __decorate([
        Column('timestamp without time zone'),
        __metadata("design:type", Object)
    ], SysPage.prototype, "change_datetime", void 0);
    __decorate([
        Column('character'),
        __metadata("design:type", Object)
    ], SysPage.prototype, "data_flag", void 0);
    SysPage = __decorate([
        Entity()
    ], SysPage);
    return SysPage;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
var entities = [
    SysCaption,
    SysCompany,
    SysDefineClass,
    SysDefineDisplay,
    SysDefineGrid,
    SysDefineGroup,
    SysDefineInput,
    SysDefineInputGrid,
    SysDefineMapping,
    SysDefineMappingControl,
    SysDefineMappingFunction,
    SysDefineSearch,
    SysDepartment,
    SysEmployee,
    SysLang,
    SysMaster,
    SysMasterDetail,
    SysMessage,
    SysModule,
    SysPage
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var CaptionService = /** @class */ (function () {
    function CaptionService(captionRepository) {
        this.captionRepository = captionRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    CaptionService.prototype.getCaptions = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.captionRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    CaptionService.prototype.getCaption = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.captionRepository.find({
                            company: context.company,
                            lang: context.lang,
                            pid: context.pid,
                            type: context.type,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    CaptionService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysCaption)),
        __metadata("design:paramtypes", [Repository])
    ], CaptionService);
    return CaptionService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var CaptionController = /** @class */ (function () {
    function CaptionController(captionService) {
        this.captionService = captionService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    CaptionController.prototype.getCaptions = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.captionService.getCaptions(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    CaptionController.prototype.getCaption = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.captionService.getCaption(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], CaptionController.prototype, "getCaptions", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], CaptionController.prototype, "getCaption", null);
    CaptionController = __decorate([
        Controller('caption'),
        __metadata("design:paramtypes", [CaptionService])
    ], CaptionController);
    return CaptionController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var CompanyService = /** @class */ (function () {
    function CompanyService(companyRepository) {
        this.companyRepository = companyRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    CompanyService.prototype.getCompanies = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.companyRepository.find({
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    CompanyService.prototype.getCompany = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.companyRepository.find({
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    CompanyService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysCompany)),
        __metadata("design:paramtypes", [Repository])
    ], CompanyService);
    return CompanyService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var CompanyController = /** @class */ (function () {
    function CompanyController(companyService) {
        this.companyService = companyService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    CompanyController.prototype.getCompanys = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.companyService.getCompanies(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    CompanyController.prototype.getCompany = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.companyService.getCompany(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], CompanyController.prototype, "getCompanys", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], CompanyController.prototype, "getCompany", null);
    CompanyController = __decorate([
        Controller('company'),
        __metadata("design:paramtypes", [CompanyService])
    ], CompanyController);
    return CompanyController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineClassService = /** @class */ (function () {
    function DefineClassService(defineClassRepository) {
        this.defineClassRepository = defineClassRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineClassService.prototype.getDefineClasses = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineClassRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineClassService.prototype.getDefineClass = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineClassRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineClassService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineClass)),
        __metadata("design:paramtypes", [Repository])
    ], DefineClassService);
    return DefineClassService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineClassController = /** @class */ (function () {
    function DefineClassController(defineClassService) {
        this.defineClassService = defineClassService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineClassController.prototype.getDefineClasses = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineClassService.getDefineClasses(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    DefineClassController.prototype.getDefineClass = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineClassService.getDefineClass(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineClassController.prototype, "getDefineClasses", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineClassController.prototype, "getDefineClass", null);
    DefineClassController = __decorate([
        Controller('define-class'),
        __metadata("design:paramtypes", [DefineClassService])
    ], DefineClassController);
    return DefineClassController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineDisplayService = /** @class */ (function () {
    function DefineDisplayService(defineDisplayRepository) {
        this.defineDisplayRepository = defineDisplayRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineDisplayService.prototype.getDefineDisplays = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineDisplayRepository.find({
                            company: context.company,
                            emp_id: context.emp_id,
                            pid: context.pid,
                            group_id: context.group_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineDisplayService.prototype.getDefineDisplay = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineDisplayRepository.find({
                            company: context.company,
                            emp_id: context.emp_id,
                            pid: context.pid,
                            group_id: context.group_id,
                            item_no: context.item_no
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineDisplayService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineDisplay)),
        __metadata("design:paramtypes", [Repository])
    ], DefineDisplayService);
    return DefineDisplayService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineDisplayController = /** @class */ (function () {
    function DefineDisplayController(defineDisplayService) {
        this.defineDisplayService = defineDisplayService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineDisplayController.prototype.getDefineDisplays = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineDisplayService.getDefineDisplays(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    DefineDisplayController.prototype.getDefineDisplay = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineDisplayService.getDefineDisplay(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineDisplayController.prototype, "getDefineDisplays", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineDisplayController.prototype, "getDefineDisplay", null);
    DefineDisplayController = __decorate([
        Controller('define-display'),
        __metadata("design:paramtypes", [DefineDisplayService])
    ], DefineDisplayController);
    return DefineDisplayController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineGridService = /** @class */ (function () {
    function DefineGridService(defineGridRepository) {
        this.defineGridRepository = defineGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineGridService.prototype.getDefineGrid = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineGridRepository.find({
                            company: context.company,
                            pid: context.pid,
                            grid_id: context.grid_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineGridService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineGrid)),
        __metadata("design:paramtypes", [Repository])
    ], DefineGridService);
    return DefineGridService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineGridController = /** @class */ (function () {
    function DefineGridController(defineGridService) {
        this.defineGridService = defineGridService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineGridController.prototype.getDefineGrid = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineGridService.getDefineGrid(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineGridController.prototype, "getDefineGrid", null);
    DefineGridController = __decorate([
        Controller('define-grid'),
        __metadata("design:paramtypes", [DefineGridService])
    ], DefineGridController);
    return DefineGridController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineGroupService = /** @class */ (function () {
    function DefineGroupService(defineGroupRepository) {
        this.defineGroupRepository = defineGroupRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineGroupService.prototype.getDefineGroups = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineGroupRepository.find({
                            company: context.company,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineGroupService.prototype.getDefineGroup = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineGroupRepository.find({
                            company: context.company,
                            pid: context.pid,
                            group_id: context.group_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineGroupService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineGroup)),
        __metadata("design:paramtypes", [Repository])
    ], DefineGroupService);
    return DefineGroupService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineGroupController = /** @class */ (function () {
    function DefineGroupController(defineGroupService) {
        this.defineGroupService = defineGroupService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineGroupController.prototype.getDefineGroups = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineGroupService.getDefineGroups(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    DefineGroupController.prototype.getDefineGroup = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineGroupService.getDefineGroup(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineGroupController.prototype, "getDefineGroups", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineGroupController.prototype, "getDefineGroup", null);
    DefineGroupController = __decorate([
        Controller('define-group'),
        __metadata("design:paramtypes", [DefineGroupService])
    ], DefineGroupController);
    return DefineGroupController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineInputService = /** @class */ (function () {
    function DefineInputService(defineInputRepository) {
        this.defineInputRepository = defineInputRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineInputService.prototype.getDefineInput = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineInputRepository.find({
                            company: context.company,
                            pid: context.pid,
                            input_id: context.input_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineInputService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineInput)),
        __metadata("design:paramtypes", [Repository])
    ], DefineInputService);
    return DefineInputService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineInputController = /** @class */ (function () {
    function DefineInputController(defineInputService) {
        this.defineInputService = defineInputService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineInputController.prototype.getDefineInput = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineInputService.getDefineInput(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineInputController.prototype, "getDefineInput", null);
    DefineInputController = __decorate([
        Controller('define-input'),
        __metadata("design:paramtypes", [DefineInputService])
    ], DefineInputController);
    return DefineInputController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineMappingControlService = /** @class */ (function () {
    function DefineMappingControlService(defineMappingControlRepository) {
        this.defineMappingControlRepository = defineMappingControlRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineMappingControlService.prototype.getDefineMappingControls = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineMappingControlRepository.find({
                            company: context.company,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DefineMappingControlService.prototype.getDefineMappingControl = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineMappingControlRepository.find({
                            company: context.company,
                            pid: context.pid,
                            item_no: context.item_no
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineMappingControlService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineMappingControl)),
        __metadata("design:paramtypes", [Repository])
    ], DefineMappingControlService);
    return DefineMappingControlService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineMappingControlController = /** @class */ (function () {
    function DefineMappingControlController(defineMappingControlService) {
        this.defineMappingControlService = defineMappingControlService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineMappingControlController.prototype.getDefineMappingControls = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineMappingControlService.getDefineMappingControls(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    DefineMappingControlController.prototype.getDefineMappingControl = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineMappingControlService.getDefineMappingControl(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineMappingControlController.prototype, "getDefineMappingControls", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineMappingControlController.prototype, "getDefineMappingControl", null);
    DefineMappingControlController = __decorate([
        Controller('define-mapping-control'),
        __metadata("design:paramtypes", [DefineMappingControlService])
    ], DefineMappingControlController);
    return DefineMappingControlController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineMappingFunctionService = /** @class */ (function () {
    function DefineMappingFunctionService(defineMappingFunctionRepository) {
        this.defineMappingFunctionRepository = defineMappingFunctionRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineMappingFunctionService.prototype.getDefineMappingFunction = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineMappingFunctionRepository.find({
                            company: context.company,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineMappingFunctionService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineMappingFunction)),
        __metadata("design:paramtypes", [Repository])
    ], DefineMappingFunctionService);
    return DefineMappingFunctionService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineMappingFunctionController = /** @class */ (function () {
    function DefineMappingFunctionController(defineMappingFunctionService) {
        this.defineMappingFunctionService = defineMappingFunctionService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineMappingFunctionController.prototype.getDefineMappingFunction = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineMappingFunctionService.getDefineMappingFunction(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineMappingFunctionController.prototype, "getDefineMappingFunction", null);
    DefineMappingFunctionController = __decorate([
        Controller('define-mapping-function'),
        __metadata("design:paramtypes", [DefineMappingFunctionService])
    ], DefineMappingFunctionController);
    return DefineMappingFunctionController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineMappingService = /** @class */ (function () {
    function DefineMappingService(defineMappingRepository) {
        this.defineMappingRepository = defineMappingRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineMappingService.prototype.getDefineMapping = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineMappingRepository.find({
                            company: context.company,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineMappingService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineMapping)),
        __metadata("design:paramtypes", [Repository])
    ], DefineMappingService);
    return DefineMappingService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineMappingController = /** @class */ (function () {
    function DefineMappingController(defineMappingService) {
        this.defineMappingService = defineMappingService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineMappingController.prototype.getDefineMapping = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineMappingService.getDefineMapping(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineMappingController.prototype, "getDefineMapping", null);
    DefineMappingController = __decorate([
        Controller('define-mapping'),
        __metadata("design:paramtypes", [DefineMappingService])
    ], DefineMappingController);
    return DefineMappingController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineSearchService = /** @class */ (function () {
    function DefineSearchService(defineSearchRepository) {
        this.defineSearchRepository = defineSearchRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineSearchService.prototype.getDefineSearch = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineSearchRepository.find({
                            company: context.company,
                            pid: context.pid,
                            search_id: context.search_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineSearchService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineSearch)),
        __metadata("design:paramtypes", [Repository])
    ], DefineSearchService);
    return DefineSearchService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineSearchController = /** @class */ (function () {
    function DefineSearchController(defineSearchService) {
        this.defineSearchService = defineSearchService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineSearchController.prototype.getDefineSearch = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineSearchService.getDefineSearch(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineSearchController.prototype, "getDefineSearch", null);
    DefineSearchController = __decorate([
        Controller('define-search'),
        __metadata("design:paramtypes", [DefineSearchService])
    ], DefineSearchController);
    return DefineSearchController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DepartmentService = /** @class */ (function () {
    function DepartmentService(departmentRepository) {
        this.departmentRepository = departmentRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DepartmentService.prototype.getDepartments = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.departmentRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    DepartmentService.prototype.getDepartment = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.departmentRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DepartmentService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDepartment)),
        __metadata("design:paramtypes", [Repository])
    ], DepartmentService);
    return DepartmentService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DepartmentController = /** @class */ (function () {
    function DepartmentController(departmentService) {
        this.departmentService = departmentService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DepartmentController.prototype.getDepartments = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.departmentService.getDepartments(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    DepartmentController.prototype.getDepartment = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.departmentService.getDepartment(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DepartmentController.prototype, "getDepartments", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DepartmentController.prototype, "getDepartment", null);
    DepartmentController = __decorate([
        Controller('department'),
        __metadata("design:paramtypes", [DepartmentService])
    ], DepartmentController);
    return DepartmentController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var EmployeeService = /** @class */ (function () {
    function EmployeeService(employeeRepository) {
        this.employeeRepository = employeeRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.getEmployees = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            var params, list, result, index, index, index, index, index, index, index, index, index, index;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        params = new Object();
                        list = [];
                        if (context.name !== undefined) {
                            params['name'] = context.name;
                        }
                        if (context.activeFlag !== undefined) {
                            params['active_flag'] = context.activeFlag;
                        }
                        if (context.address1 !== undefined) {
                            params['address1'] = context.address1;
                        }
                        if (context.address2 !== undefined) {
                            params['address2'] = context.address2;
                        }
                        if (context.tel1 !== undefined) {
                            params['tel1'] = context.tel1;
                        }
                        if (context.tel2 !== undefined) {
                            params['tel2'] = context.tel2;
                        }
                        if (context.email !== undefined) {
                            params['email'] = context.email;
                        }
                        if (context.sortNo !== undefined) {
                            params['sort_no'] = context.sortNo;
                        }
                        if (context.changeCount !== undefined) {
                            params['change_count'] = context.changeCount;
                        }
                        if (context.dataFlag !== undefined) {
                            params['data_flag'] = context.dataFlag;
                        }
                        return [4 /*yield*/, this.employeeRepository.find(__assign({ company: context.company, lang: context.lang }, params))];
                    case 1:
                        result = _a.sent();
                        list = __spread(result);
                        // NOTE Emp code
                        if (context.codeFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.code === context.codeFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(index);
                            }
                        }
                        if (context.codeTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.code === context.codeTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE create_emp_id
                        if (context.createEmpFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_emp_id === context.createEmpFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.createEmpTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_emp_id === context.createEmpTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE create_datetime
                        if (context.createDatetimeFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_datetime === new Date(context.createDatetimeFrom);
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.createDatetimeTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.create_datetime === context.createDatetimeTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE change_emp_id
                        if (context.changeEmpFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_emp_id === context.changeEmpFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.changeEmpTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_emp_id === context.changeEmpTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        // NOTE change_datetime
                        if (context.changeDatetimeFrom !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_datetime === context.changeDatetimeFrom;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        if (context.changeDatetimeTo !== undefined) {
                            index = list.findIndex((/**
                             * @param {?} element
                             * @return {?}
                             */
                            function (element) {
                                return element.change_datetime === context.changeDatetimeTo;
                            }));
                            if (index >= 0) {
                                list = list.slice(0, index + 1);
                            }
                        }
                        return [2 /*return*/, list];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.getEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.employeeRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.addEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            var result, contextVi, contextEn, contextJa;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!(context.code !== '')) return [3 /*break*/, 6];
                        return [4 /*yield*/, this.employeeRepository.find({
                                company: context.company,
                                code: context.code
                            })];
                    case 1:
                        result = _a.sent();
                        if (!(result.length <= 0)) return [3 /*break*/, 5];
                        contextVi = __assign({}, context);
                        contextVi.lang = 'vi-VN';
                        contextEn = __assign({}, context);
                        contextEn.lang = 'en-US';
                        contextJa = __assign({}, context);
                        contextJa.lang = 'ja-JP';
                        return [4 /*yield*/, this.employeeRepository.save(contextVi)];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.save(contextEn)];
                    case 3:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.save(contextJa)];
                    case 4:
                        _a.sent();
                        return [2 /*return*/, true];
                    case 5: return [2 /*return*/, false];
                    case 6: return [2 /*return*/, false];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.updateEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            var result, contextVi, contextEn, contextJa;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!(context.code !== '')) return [3 /*break*/, 6];
                        return [4 /*yield*/, this.employeeRepository.find({
                                company: context.company,
                                code: context.code
                            })];
                    case 1:
                        result = _a.sent();
                        if (!(result.length > 0)) return [3 /*break*/, 5];
                        context.change_count++;
                        context.change_datetime = null;
                        contextVi = __assign({}, context);
                        contextVi.lang = 'vi-VN';
                        contextEn = __assign({}, context);
                        contextEn.lang = 'en-US';
                        contextJa = __assign({}, context);
                        contextJa.lang = 'ja-JP';
                        return [4 /*yield*/, this.employeeRepository.update([contextVi.company, contextVi.lang, contextVi.code], contextVi)];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.update([contextEn.company, contextEn.lang, contextEn.code], contextEn)];
                    case 3:
                        _a.sent();
                        return [4 /*yield*/, this.employeeRepository.update([contextJa.company, contextJa.lang, contextJa.code], contextJa)];
                    case 4:
                        _a.sent();
                        return [2 /*return*/, true];
                    case 5: return [2 /*return*/, false];
                    case 6: return [2 /*return*/, false];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.deleteEmployee = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            var result;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.employeeRepository.delete({ code: context })];
                    case 1:
                        result = _a.sent();
                        if (result.affected > 0) {
                            return [2 /*return*/, true];
                        }
                        return [2 /*return*/, false];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    EmployeeService.prototype.deleteEmployees = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            var result;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.employeeRepository.delete(context)];
                    case 1:
                        result = _a.sent();
                        if (result.affected > 0) {
                            return [2 /*return*/, true];
                        }
                        return [2 /*return*/, false];
                }
            });
        });
    };
    EmployeeService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysEmployee)),
        __metadata("design:paramtypes", [Repository])
    ], EmployeeService);
    return EmployeeService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var EmployeeController = /** @class */ (function () {
    function EmployeeController(employeeService) {
        this.employeeService = employeeService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    EmployeeController.prototype.getEmployees = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.employeeService.getEmployees(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    EmployeeController.prototype.getEmployee = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.employeeService.getEmployee(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    EmployeeController.prototype.addEmployee = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.employeeService.addEmployee(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    EmployeeController.prototype.updateEmployee = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.employeeService.updateEmployee(body);
    };
    /**
     * @param {?} params
     * @return {?}
     */
    EmployeeController.prototype.deleteEmployee = /**
     * @param {?} params
     * @return {?}
     */
    function (params) {
        return this.employeeService.deleteEmployee(params.id);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    EmployeeController.prototype.deleteEmployees = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.employeeService.deleteEmployees(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], EmployeeController.prototype, "getEmployees", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], EmployeeController.prototype, "getEmployee", null);
    __decorate([
        Post('add'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], EmployeeController.prototype, "addEmployee", null);
    __decorate([
        Post('update'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], EmployeeController.prototype, "updateEmployee", null);
    __decorate([
        Delete(':id'),
        __param(0, Param()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], EmployeeController.prototype, "deleteEmployee", null);
    __decorate([
        Post('delete/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], EmployeeController.prototype, "deleteEmployees", null);
    EmployeeController = __decorate([
        Controller('employee'),
        __metadata("design:paramtypes", [EmployeeService])
    ], EmployeeController);
    return EmployeeController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var LangService = /** @class */ (function () {
    function LangService(langRepository) {
        this.langRepository = langRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    LangService.prototype.getLangs = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.langRepository.find({
                        company: context.company
                    })];
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    LangService.prototype.getLang = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.langRepository.find({
                        company: context.company,
                        code: context.code
                    })];
            });
        });
    };
    LangService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysLang)),
        __metadata("design:paramtypes", [Repository])
    ], LangService);
    return LangService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var LangController = /** @class */ (function () {
    function LangController(langService) {
        this.langService = langService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    LangController.prototype.getLangs = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.langService.getLangs(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    LangController.prototype.getLang = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.langService.getLang(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], LangController.prototype, "getLangs", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], LangController.prototype, "getLang", null);
    LangController = __decorate([
        Controller('lang'),
        __metadata("design:paramtypes", [LangService])
    ], LangController);
    return LangController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MasterDetailService = /** @class */ (function () {
    function MasterDetailService(masterDetailRepository) {
        this.masterDetailRepository = masterDetailRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    MasterDetailService.prototype.getMasterDetails = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterDetailRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code,
                            master_code: context.master_code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    MasterDetailService.prototype.getMasterDetail = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterDetailRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code,
                            master_code: context.master_code,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    MasterDetailService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysMasterDetail)),
        __metadata("design:paramtypes", [Repository])
    ], MasterDetailService);
    return MasterDetailService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MasterDetailController = /** @class */ (function () {
    function MasterDetailController(masterDetailService) {
        this.masterDetailService = masterDetailService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    MasterDetailController.prototype.getMasterDetails = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.masterDetailService.getMasterDetails(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    MasterDetailController.prototype.getMasterDetail = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.masterDetailService.getMasterDetail(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], MasterDetailController.prototype, "getMasterDetails", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], MasterDetailController.prototype, "getMasterDetail", null);
    MasterDetailController = __decorate([
        Controller('master-detail'),
        __metadata("design:paramtypes", [MasterDetailService])
    ], MasterDetailController);
    return MasterDetailController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MasterService = /** @class */ (function () {
    function MasterService(masterRepository) {
        this.masterRepository = masterRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    MasterService.prototype.getMasters = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    MasterService.prototype.getMaster = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.masterRepository.find({
                            company: context.company,
                            lang: context.lang,
                            class_code: context.class_code,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    MasterService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysMaster)),
        __metadata("design:paramtypes", [Repository])
    ], MasterService);
    return MasterService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MasterController = /** @class */ (function () {
    function MasterController(masterService) {
        this.masterService = masterService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    MasterController.prototype.getMasters = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.masterService.getMasters(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    MasterController.prototype.getMaster = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.masterService.getMaster(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], MasterController.prototype, "getMasters", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], MasterController.prototype, "getMaster", null);
    MasterController = __decorate([
        Controller('master'),
        __metadata("design:paramtypes", [MasterService])
    ], MasterController);
    return MasterController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MessageService = /** @class */ (function () {
    function MessageService(messageRepository) {
        this.messageRepository = messageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    MessageService.prototype.getMessage = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.messageRepository.find({
                            company: context.company,
                            lang: context.lang,
                            code: context.code
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    MessageService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysMessage)),
        __metadata("design:paramtypes", [Repository])
    ], MessageService);
    return MessageService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MessageController = /** @class */ (function () {
    function MessageController(messageService) {
        this.messageService = messageService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    MessageController.prototype.getMessage = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.messageService.getMessage(body);
    };
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], MessageController.prototype, "getMessage", null);
    MessageController = __decorate([
        Controller('message'),
        __metadata("design:paramtypes", [MessageService])
    ], MessageController);
    return MessageController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var ModuleService = /** @class */ (function () {
    function ModuleService(moduleRepository) {
        this.moduleRepository = moduleRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    ModuleService.prototype.getModules = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.moduleRepository.find({
                            company: context.company,
                            lang: context.lang
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    ModuleService.prototype.getModule = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.moduleRepository.find({
                            company: context.company,
                            lang: context.lang,
                            mid: context.mid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    ModuleService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysModule)),
        __metadata("design:paramtypes", [Repository])
    ], ModuleService);
    return ModuleService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var ModuleController = /** @class */ (function () {
    function ModuleController(moduleService) {
        this.moduleService = moduleService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    ModuleController.prototype.getModules = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.moduleService.getModules(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    ModuleController.prototype.getModule = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.moduleService.getModule(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], ModuleController.prototype, "getModules", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], ModuleController.prototype, "getModule", null);
    ModuleController = __decorate([
        Controller('module'),
        __metadata("design:paramtypes", [ModuleService])
    ], ModuleController);
    return ModuleController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var PageService = /** @class */ (function () {
    function PageService(pageRepository) {
        this.pageRepository = pageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    PageService.prototype.getPages = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.pageRepository.find({
                            company: context.company,
                            lang: context.lang,
                            mid: context.mid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} context
     * @return {?}
     */
    PageService.prototype.getPage = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.pageRepository.find({
                            company: context.company,
                            lang: context.lang,
                            mid: context.mid,
                            pid: context.pid
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    PageService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysPage)),
        __metadata("design:paramtypes", [Repository])
    ], PageService);
    return PageService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var PageController = /** @class */ (function () {
    function PageController(pageService) {
        this.pageService = pageService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    PageController.prototype.getPages = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.pageService.getPages(body);
    };
    /**
     * @param {?} body
     * @return {?}
     */
    PageController.prototype.getPage = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.pageService.getPage(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], PageController.prototype, "getPages", null);
    __decorate([
        Post('get/one'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], PageController.prototype, "getPage", null);
    PageController = __decorate([
        Controller('page'),
        __metadata("design:paramtypes", [PageService])
    ], PageController);
    return PageController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineInputGridService = /** @class */ (function () {
    function DefineInputGridService(defineInputGridRepository) {
        this.defineInputGridRepository = defineInputGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    DefineInputGridService.prototype.getDefineInputGrid = /**
     * @param {?} context
     * @return {?}
     */
    function (context) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.defineInputGridRepository.find({
                            company: context.company,
                            pid: context.pid,
                            input_grid_id: context.input_grid_id
                        })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DefineInputGridService = __decorate([
        Injectable(),
        __param(0, InjectRepository(SysDefineInputGrid)),
        __metadata("design:paramtypes", [Repository])
    ], DefineInputGridService);
    return DefineInputGridService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DefineInputGridController = /** @class */ (function () {
    function DefineInputGridController(defineInputGridService) {
        this.defineInputGridService = defineInputGridService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    DefineInputGridController.prototype.getDefineInputGrid = /**
     * @param {?} body
     * @return {?}
     */
    function (body) {
        return this.defineInputGridService.getDefineInputGrid(body);
    };
    __decorate([
        Post('get/many'),
        __param(0, Body()),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], DefineInputGridController.prototype, "getDefineInputGrid", null);
    DefineInputGridController = __decorate([
        Controller('define-input-grid'),
        __metadata("design:paramtypes", [DefineInputGridService])
    ], DefineInputGridController);
    return DefineInputGridController;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
var controllers = [
    CaptionController,
    CompanyController,
    DefineClassController,
    DefineDisplayController,
    DefineGridController,
    DefineGroupController,
    DefineInputGridController,
    DefineInputController,
    DefineMappingControlController,
    DefineMappingFunctionController,
    DefineMappingController,
    DefineSearchController,
    DepartmentController,
    EmployeeController,
    LangController,
    MasterDetailController,
    MasterController,
    MessageController,
    ModuleController,
    PageController
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
var services = [
    CaptionService,
    CompanyService,
    DefineClassService,
    DefineDisplayService,
    DefineGridService,
    DefineGroupService,
    DefineInputGridService,
    DefineInputService,
    DefineMappingControlService,
    DefineMappingFunctionService,
    DefineMappingService,
    DefineSearchService,
    DepartmentService,
    EmployeeService,
    LangService,
    MasterDetailService,
    MasterService,
    MessageService,
    ModuleService,
    PageService
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var AitCoreModule = /** @class */ (function () {
    function AitCoreModule() {
    }
    AitCoreModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        RouterModule.forChild([
                        /* {path: '', pathMatch: 'full', component: InsertYourComponentHere} */
                        ])
                    ]
                },] }
    ];
    return AitCoreModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { AitCoreModule, controllers, dbConfig, entities, services, SysCaption as a, SysCompany as b, DefineDisplayController as ba, DefineDisplayService as bb, DefineGridController as bc, DefineGridService as bd, DefineGroupController as be, DefineGroupService as bf, DefineInputGridController as bg, DefineInputGridService as bh, DefineInputController as bi, DefineInputService as bj, DefineMappingControlController as bk, DefineMappingControlService as bl, DefineMappingFunctionController as bm, DefineMappingFunctionService as bn, DefineMappingController as bo, DefineMappingService as bp, DefineSearchController as bq, DefineSearchService as br, DepartmentController as bs, DepartmentService as bt, EmployeeController as bu, EmployeeService as bv, LangController as bw, LangService as bx, MasterDetailController as by, MasterDetailService as bz, SysDefineClass as c, MasterController as ca, MasterService as cb, MessageController as cc, MessageService as cd, ModuleController as ce, ModuleService as cf, PageController as cg, PageService as ch, SysDefineDisplay as d, SysDefineGrid as e, SysDefineGroup as f, SysDefineInput as g, SysDefineInputGrid as h, SysDefineMapping as i, SysDefineMappingControl as j, SysDefineMappingFunction as k, SysDefineSearch as l, SysDepartment as m, SysEmployee as n, SysLang as o, SysMaster as p, SysMasterDetail as q, SysMessage as r, SysModule as s, SysPage as t, CaptionController as u, CaptionService as v, CompanyController as w, CompanyService as x, DefineClassController as y, DefineClassService as z };
