// @aureole/core/esm2015/lib/controllers/employee.controller.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Body, Controller, Delete, Param, Post } from '@nestjs/common';
import { EmployeeService } from '../services/employee.service';
let EmployeeController = class EmployeeController {
    /**
     * @param {?} employeeService
     */
    constructor(employeeService) {
        this.employeeService = employeeService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getEmployees(body) {
        return this.employeeService.getEmployees(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getEmployee(body) {
        return this.employeeService.getEmployee(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    addEmployee(body) {
        return this.employeeService.addEmployee(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    updateEmployee(body) {
        return this.employeeService.updateEmployee(body);
    }
    /**
     * @param {?} params
     * @return {?}
     */
    deleteEmployee(params) {
        return this.employeeService.deleteEmployee(params.id);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    deleteEmployees(body) {
        return this.employeeService.deleteEmployees(body);
    }
};
tslib_1.__decorate([
    Post('get/many'),
    tslib_1.__param(0, Body()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object]),
    tslib_1.__metadata("design:returntype", void 0)
], EmployeeController.prototype, "getEmployees", null);
tslib_1.__decorate([
    Post('get/one'),
    tslib_1.__param(0, Body()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object]),
    tslib_1.__metadata("design:returntype", void 0)
], EmployeeController.prototype, "getEmployee", null);
tslib_1.__decorate([
    Post('add'),
    tslib_1.__param(0, Body()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object]),
    tslib_1.__metadata("design:returntype", void 0)
], EmployeeController.prototype, "addEmployee", null);
tslib_1.__decorate([
    Post('update'),
    tslib_1.__param(0, Body()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object]),
    tslib_1.__metadata("design:returntype", void 0)
], EmployeeController.prototype, "updateEmployee", null);
tslib_1.__decorate([
    Delete(':id'),
    tslib_1.__param(0, Param()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object]),
    tslib_1.__metadata("design:returntype", void 0)
], EmployeeController.prototype, "deleteEmployee", null);
tslib_1.__decorate([
    Post('delete/many'),
    tslib_1.__param(0, Body()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object]),
    tslib_1.__metadata("design:returntype", void 0)
], EmployeeController.prototype, "deleteEmployees", null);
EmployeeController = tslib_1.__decorate([
    Controller('employee'),
    tslib_1.__metadata("design:paramtypes", [EmployeeService])
], EmployeeController);
export { EmployeeController };
if (false) {
    /**
     * @type {?}
     * @private
     */
    EmployeeController.prototype.employeeService;
}
