// @aureole/core/esm5/lib/ait-core.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
export { dbConfig } from './configs/typeorm.config';
export { entities } from './configs/entities.config';
export { controllers } from './configs/controllers.config';
export { services } from './configs/services.config';
var AitCoreModule = /** @class */ (function () {
    function AitCoreModule() {
    }
    AitCoreModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        RouterModule.forChild([
                        /* {path: '', pathMatch: 'full', component: InsertYourComponentHere} */
                        ])
                    ]
                },] }
    ];
    return AitCoreModule;
}());
export { AitCoreModule };
