// @nrwl/schematics/src/lib-versions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.angularCliVersion = '7.3.1';
exports.angularVersion = '^7.0.0';
exports.angularDevkitVersion = '0.13.1';
exports.angularJsVersion = '1.6.6';
exports.ngrxVersion = '7.2.0';
exports.ngrxStoreFreezeVersion = '0.2.4';
exports.nxVersion = '7.8.5';
exports.schematicsVersion = '7.8.5';
exports.latestMigration = '20180507-create-nx-json';
exports.prettierVersion = '1.16.4';
exports.typescriptVersion = '3.2.2';
exports.rxjsVersion = '6.3.3';
exports.jestVersion = '24.1.0';
exports.jestTypesVersion = '24.0.9';
exports.jestPresetAngularVersion = '7.0.0';
exports.tsJestversion = '24.0.0';
exports.jasmineMarblesVersion = '0.4.0';
exports.cypressVersion = '3.1.0';
exports.expressVersion = '4.16.3';
exports.expressTypingsVersion = '4.16.0';
exports.nestJsVersion = '5.5.0';
exports.nestJsSchematicsVersion = '5.11.2';
exports.reactVersions = {
    framework: '16.8.3',
    reactTypes: '16.8.4',
    reactDomTypes: '16.8.2',
    testingLibrary: '6.0.0'
};
exports.documentRegisterElementVersion = '1.13.1';
exports.libVersions = {
    angularVersion: exports.angularVersion,
    angularDevkitVersion: exports.angularDevkitVersion,
    angularCliVersion: exports.angularCliVersion,
    angularJsVersion: exports.angularJsVersion,
    ngrxVersion: exports.ngrxVersion,
    ngrxStoreFreezeVersion: exports.ngrxStoreFreezeVersion,
    nxVersion: exports.nxVersion,
    schematicsVersion: exports.schematicsVersion,
    prettierVersion: exports.prettierVersion,
    latestMigration: exports.latestMigration,
    typescriptVersion: exports.typescriptVersion,
    rxjsVersion: exports.rxjsVersion,
    jasmineMarblesVersion: exports.jasmineMarblesVersion
};
