// @nrwl/builders/src/utils/webpack/node.config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var webpack_1 = require("webpack");
var mergeWebpack = require("webpack-merge");
var nodeExternals = require("webpack-node-externals");
var config_1 = require("./config");
function getNodePartial(options) {
    var webpackConfig = {
        output: {
            libraryTarget: 'commonjs'
        },
        target: 'node',
        node: false
    };
    if (options.optimization) {
        webpackConfig.optimization = {
            minimize: false,
            concatenateModules: false
        };
    }
    if (options.externalDependencies === 'all') {
        webpackConfig.externals = [nodeExternals()];
    }
    else if (Array.isArray(options.externalDependencies)) {
        webpackConfig.externals = [
            function (context, request, callback) {
                if (options.externalDependencies.includes(request)) {
                    // not bundled
                    return callback(null, 'commonjs ' + request);
                }
                // bundled
                callback();
            }
        ];
    }
    if (options.sourceMap) {
        webpackConfig.plugins = [
            new webpack_1.BannerPlugin({
                banner: 'require("source-map-support").install();',
                raw: true,
                entryOnly: false
            })
        ];
    }
    return webpackConfig;
}
function getNodeWebpackConfig(options) {
    return mergeWebpack([
        config_1.getBaseWebpackPartial(options),
        getNodePartial(options)
    ]);
}
exports.getNodeWebpackConfig = getNodeWebpackConfig;
