// @aureole/core/fesm2015/ait-ait-core.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { __decorate, __metadata, __awaiter, __param } from 'tslib';
import { PrimaryColumn, Column, Entity, Repository } from 'typeorm';
import { Injectable, Post, Body, Controller, Delete, Param } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
const dbConfig = {
    type: 'postgres',
    host: '192.168.136.12',
    port: 1102,
    username: 'ait',
    password: 'ait',
    database: 'ait',
    schema: 'ait_pm_dev',
    synchronize: true
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysCaption = class SysCaption {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCaption.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCaption.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCaption.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysCaption.prototype, "type", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCaption.prototype, "code", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysCaption.prototype, "name", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysCaption.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCaption.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysCaption.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCaption.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysCaption.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysCaption.prototype, "data_flag", void 0);
SysCaption = __decorate([
    Entity()
], SysCaption);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysCompany = class SysCompany {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysCompany.prototype, "active_flag", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "address1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "address2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "tel1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "tel2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "email", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysCompany.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysCompany.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysCompany.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysCompany.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysCompany.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysCompany.prototype, "data_flag", void 0);
SysCompany = __decorate([
    Entity()
], SysCompany);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineClass = class SysDefineClass {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "active_flag", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "remark", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "sort_no", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "table_ref", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineClass.prototype, "data_flag", void 0);
SysDefineClass = __decorate([
    Entity()
], SysDefineClass);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineDisplay = class SysDefineDisplay {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "emp_id", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "group_id", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "item_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "disp_no", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "item_disp", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "width", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "height", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "row_start", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "row_end", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "col_start", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "col_end", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "color_bg", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "color_text", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "class", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "stype", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineDisplay.prototype, "data_flag", void 0);
SysDefineDisplay = __decorate([
    Entity()
], SysDefineDisplay);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineGrid = class SysDefineGrid {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "grid_id", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "item_no", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "data_type", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 100 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "data_format", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "cap_type", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "cap_code", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineGrid.prototype, "data_flag", void 0);
SysDefineGrid = __decorate([
    Entity()
], SysDefineGrid);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineGroup = class SysDefineGroup {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "group_id", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "cap_type", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "cap_code", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineGroup.prototype, "data_flag", void 0);
SysDefineGroup = __decorate([
    Entity()
], SysDefineGroup);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineInput = class SysDefineInput {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "input_id", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "item_no", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "input_type", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "cap_type", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "cap_code", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "default_value", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineInput.prototype, "data_flag", void 0);
SysDefineInput = __decorate([
    Entity()
], SysDefineInput);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineInputGrid = class SysDefineInputGrid {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "input_grid_id", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "item_no", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "input_type", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "cap_type", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "cap_code", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "default_value", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineInputGrid.prototype, "data_flag", void 0);
SysDefineInputGrid = __decorate([
    Entity()
], SysDefineInputGrid);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineMapping = class SysDefineMapping {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "mapping_group_id", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "mapping_item_no", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 250 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "mapping_item_field", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 50 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "mapping_table_name", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 50 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "mapping_column_name", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "mapping_data_type", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineMapping.prototype, "data_flag", void 0);
SysDefineMapping = __decorate([
    Entity()
], SysDefineMapping);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineMappingControl = class SysDefineMappingControl {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "item_no", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "control_type", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_table", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column0", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column3", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column4", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column5", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column6", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column7", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column8", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "mapping_column9", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineMappingControl.prototype, "data_flag", void 0);
SysDefineMappingControl = __decorate([
    Entity()
], SysDefineMappingControl);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineMappingFunction = class SysDefineMappingFunction {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "screen_type", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "function_type", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "active_flag", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineMappingFunction.prototype, "data_flag", void 0);
SysDefineMappingFunction = __decorate([
    Entity()
], SysDefineMappingFunction);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDefineSearch = class SysDefineSearch {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "pid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "search_id", void 0);
__decorate([
    PrimaryColumn('integer'),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "item_no", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "search_type", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "cap_type", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "cap_code", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "default_value", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDefineSearch.prototype, "data_flag", void 0);
SysDefineSearch = __decorate([
    Entity()
], SysDefineSearch);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysDepartment = class SysDepartment {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysDepartment.prototype, "active_flag", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "address1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "address2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "tel1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "tel2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "email", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDepartment.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysDepartment.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDepartment.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysDepartment.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysDepartment.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysDepartment.prototype, "data_flag", void 0);
SysDepartment = __decorate([
    Entity()
], SysDepartment);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysEmployee = class SysEmployee {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysEmployee.prototype, "active_flag", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "address1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 250 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "address2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "tel1", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "tel2", void 0);
__decorate([
    Column('character varying', { nullable: true, length: 50 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "email", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysEmployee.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysEmployee.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysEmployee.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysEmployee.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysEmployee.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysEmployee.prototype, "data_flag", void 0);
SysEmployee = __decorate([
    Entity()
], SysEmployee);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysLang = class SysLang {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysLang.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysLang.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysLang.prototype, "name", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysLang.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysLang.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysLang.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysLang.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysLang.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysLang.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysLang.prototype, "data_flag", void 0);
SysLang = __decorate([
    Entity()
], SysLang);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysMaster = class SysMaster {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "class_code", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysMaster.prototype, "active_flag", void 0);
__decorate([
    Column('text', { nullable: true }),
    __metadata("design:type", Object)
], SysMaster.prototype, "remark", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysMaster.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysMaster.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysMaster.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMaster.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysMaster.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysMaster.prototype, "data_flag", void 0);
SysMaster = __decorate([
    Entity()
], SysMaster);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysMasterDetail = class SysMasterDetail {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "class_code", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "master_code", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "code", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "active_flag", void 0);
__decorate([
    Column('text', { nullable: true }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "remark", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysMasterDetail.prototype, "data_flag", void 0);
SysMasterDetail = __decorate([
    Entity()
], SysMasterDetail);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysMessage = class SysMessage {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMessage.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMessage.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMessage.prototype, "code", void 0);
__decorate([
    Column('text'),
    __metadata("design:type", Object)
], SysMessage.prototype, "message", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysMessage.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMessage.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysMessage.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysMessage.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysMessage.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysMessage.prototype, "data_flag", void 0);
SysMessage = __decorate([
    Entity()
], SysMessage);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysModule = class SysModule {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "mid", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysModule.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysModule.prototype, "active_flag", void 0);
__decorate([
    Column('character varying', { length: 250 }),
    __metadata("design:type", Object)
], SysModule.prototype, "icon", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "slug", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "pid", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysModule.prototype, "sort_no", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysModule.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysModule.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysModule.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysModule.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysModule.prototype, "data_flag", void 0);
SysModule = __decorate([
    Entity()
], SysModule);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let SysPage = class SysPage {
};
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "company", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "lang", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "mid", void 0);
__decorate([
    PrimaryColumn('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "pid", void 0);
__decorate([
    Column('character varying', { length: 100 }),
    __metadata("design:type", Object)
], SysPage.prototype, "name", void 0);
__decorate([
    Column('boolean'),
    __metadata("design:type", Object)
], SysPage.prototype, "active_flag", void 0);
__decorate([
    Column('character varying', { length: 1000 }),
    __metadata("design:type", Object)
], SysPage.prototype, "description", void 0);
__decorate([
    Column('character varying', { length: 250 }),
    __metadata("design:type", Object)
], SysPage.prototype, "icon", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "slug", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysPage.prototype, "sort_no", void 0);
__decorate([
    Column('character varying', { length: 50 }),
    __metadata("design:type", Object)
], SysPage.prototype, "mapping_table_name", void 0);
__decorate([
    Column('character varying', { length: 50 }),
    __metadata("design:type", Object)
], SysPage.prototype, "mapping_entity_name", void 0);
__decorate([
    Column('integer'),
    __metadata("design:type", Object)
], SysPage.prototype, "change_count", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "create_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysPage.prototype, "create_datetime", void 0);
__decorate([
    Column('character varying', { length: 20 }),
    __metadata("design:type", Object)
], SysPage.prototype, "change_emp_id", void 0);
__decorate([
    Column('timestamp without time zone'),
    __metadata("design:type", Object)
], SysPage.prototype, "change_datetime", void 0);
__decorate([
    Column('character'),
    __metadata("design:type", Object)
], SysPage.prototype, "data_flag", void 0);
SysPage = __decorate([
    Entity()
], SysPage);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
const entities = [
    SysCaption,
    SysCompany,
    SysDefineClass,
    SysDefineDisplay,
    SysDefineGrid,
    SysDefineGroup,
    SysDefineInput,
    SysDefineInputGrid,
    SysDefineMapping,
    SysDefineMappingControl,
    SysDefineMappingFunction,
    SysDefineSearch,
    SysDepartment,
    SysEmployee,
    SysLang,
    SysMaster,
    SysMasterDetail,
    SysMessage,
    SysModule,
    SysPage
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let CaptionService = class CaptionService {
    /**
     * @param {?} captionRepository
     */
    constructor(captionRepository) {
        this.captionRepository = captionRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCaptions(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.captionRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCaption(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.captionRepository.find({
                company: context.company,
                lang: context.lang,
                pid: context.pid,
                type: context.type,
                code: context.code
            });
        });
    }
};
CaptionService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysCaption)),
    __metadata("design:paramtypes", [Repository])
], CaptionService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let CaptionController = class CaptionController {
    /**
     * @param {?} captionService
     */
    constructor(captionService) {
        this.captionService = captionService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getCaptions(body) {
        return this.captionService.getCaptions(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getCaption(body) {
        return this.captionService.getCaption(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], CaptionController.prototype, "getCaptions", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], CaptionController.prototype, "getCaption", null);
CaptionController = __decorate([
    Controller('caption'),
    __metadata("design:paramtypes", [CaptionService])
], CaptionController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let CompanyService = class CompanyService {
    /**
     * @param {?} companyRepository
     */
    constructor(companyRepository) {
        this.companyRepository = companyRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCompanies(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.companyRepository.find({
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getCompany(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.companyRepository.find({
                lang: context.lang,
                code: context.code
            });
        });
    }
};
CompanyService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysCompany)),
    __metadata("design:paramtypes", [Repository])
], CompanyService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let CompanyController = class CompanyController {
    /**
     * @param {?} companyService
     */
    constructor(companyService) {
        this.companyService = companyService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getCompanys(body) {
        return this.companyService.getCompanies(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getCompany(body) {
        return this.companyService.getCompany(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], CompanyController.prototype, "getCompanys", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], CompanyController.prototype, "getCompany", null);
CompanyController = __decorate([
    Controller('company'),
    __metadata("design:paramtypes", [CompanyService])
], CompanyController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineClassService = class DefineClassService {
    /**
     * @param {?} defineClassRepository
     */
    constructor(defineClassRepository) {
        this.defineClassRepository = defineClassRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineClasses(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineClassRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineClass(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineClassRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
};
DefineClassService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineClass)),
    __metadata("design:paramtypes", [Repository])
], DefineClassService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineClassController = class DefineClassController {
    /**
     * @param {?} defineClassService
     */
    constructor(defineClassService) {
        this.defineClassService = defineClassService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineClasses(body) {
        return this.defineClassService.getDefineClasses(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineClass(body) {
        return this.defineClassService.getDefineClass(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineClassController.prototype, "getDefineClasses", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineClassController.prototype, "getDefineClass", null);
DefineClassController = __decorate([
    Controller('define-class'),
    __metadata("design:paramtypes", [DefineClassService])
], DefineClassController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineDisplayService = class DefineDisplayService {
    /**
     * @param {?} defineDisplayRepository
     */
    constructor(defineDisplayRepository) {
        this.defineDisplayRepository = defineDisplayRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineDisplays(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineDisplayRepository.find({
                company: context.company,
                emp_id: context.emp_id,
                pid: context.pid,
                group_id: context.group_id
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineDisplay(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineDisplayRepository.find({
                company: context.company,
                emp_id: context.emp_id,
                pid: context.pid,
                group_id: context.group_id,
                item_no: context.item_no
            });
        });
    }
};
DefineDisplayService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineDisplay)),
    __metadata("design:paramtypes", [Repository])
], DefineDisplayService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineDisplayController = class DefineDisplayController {
    /**
     * @param {?} defineDisplayService
     */
    constructor(defineDisplayService) {
        this.defineDisplayService = defineDisplayService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineDisplays(body) {
        return this.defineDisplayService.getDefineDisplays(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineDisplay(body) {
        return this.defineDisplayService.getDefineDisplay(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineDisplayController.prototype, "getDefineDisplays", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineDisplayController.prototype, "getDefineDisplay", null);
DefineDisplayController = __decorate([
    Controller('define-display'),
    __metadata("design:paramtypes", [DefineDisplayService])
], DefineDisplayController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineGridService = class DefineGridService {
    /**
     * @param {?} defineGridRepository
     */
    constructor(defineGridRepository) {
        this.defineGridRepository = defineGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineGrid(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineGridRepository.find({
                company: context.company,
                pid: context.pid,
                grid_id: context.grid_id
            });
        });
    }
};
DefineGridService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineGrid)),
    __metadata("design:paramtypes", [Repository])
], DefineGridService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineGridController = class DefineGridController {
    /**
     * @param {?} defineGridService
     */
    constructor(defineGridService) {
        this.defineGridService = defineGridService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineGrid(body) {
        return this.defineGridService.getDefineGrid(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineGridController.prototype, "getDefineGrid", null);
DefineGridController = __decorate([
    Controller('define-grid'),
    __metadata("design:paramtypes", [DefineGridService])
], DefineGridController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineGroupService = class DefineGroupService {
    /**
     * @param {?} defineGroupRepository
     */
    constructor(defineGroupRepository) {
        this.defineGroupRepository = defineGroupRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineGroups(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineGroupRepository.find({
                company: context.company,
                pid: context.pid
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineGroup(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineGroupRepository.find({
                company: context.company,
                pid: context.pid,
                group_id: context.group_id
            });
        });
    }
};
DefineGroupService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineGroup)),
    __metadata("design:paramtypes", [Repository])
], DefineGroupService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineGroupController = class DefineGroupController {
    /**
     * @param {?} defineGroupService
     */
    constructor(defineGroupService) {
        this.defineGroupService = defineGroupService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineGroups(body) {
        return this.defineGroupService.getDefineGroups(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineGroup(body) {
        return this.defineGroupService.getDefineGroup(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineGroupController.prototype, "getDefineGroups", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineGroupController.prototype, "getDefineGroup", null);
DefineGroupController = __decorate([
    Controller('define-group'),
    __metadata("design:paramtypes", [DefineGroupService])
], DefineGroupController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineInputService = class DefineInputService {
    /**
     * @param {?} defineInputRepository
     */
    constructor(defineInputRepository) {
        this.defineInputRepository = defineInputRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineInput(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineInputRepository.find({
                company: context.company,
                pid: context.pid,
                input_id: context.input_id
            });
        });
    }
};
DefineInputService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineInput)),
    __metadata("design:paramtypes", [Repository])
], DefineInputService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineInputController = class DefineInputController {
    /**
     * @param {?} defineInputService
     */
    constructor(defineInputService) {
        this.defineInputService = defineInputService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineInput(body) {
        return this.defineInputService.getDefineInput(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineInputController.prototype, "getDefineInput", null);
DefineInputController = __decorate([
    Controller('define-input'),
    __metadata("design:paramtypes", [DefineInputService])
], DefineInputController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineMappingControlService = class DefineMappingControlService {
    /**
     * @param {?} defineMappingControlRepository
     */
    constructor(defineMappingControlRepository) {
        this.defineMappingControlRepository = defineMappingControlRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineMappingControls(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineMappingControlRepository.find({
                company: context.company,
                pid: context.pid
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineMappingControl(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineMappingControlRepository.find({
                company: context.company,
                pid: context.pid,
                item_no: context.item_no
            });
        });
    }
};
DefineMappingControlService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineMappingControl)),
    __metadata("design:paramtypes", [Repository])
], DefineMappingControlService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineMappingControlController = class DefineMappingControlController {
    /**
     * @param {?} defineMappingControlService
     */
    constructor(defineMappingControlService) {
        this.defineMappingControlService = defineMappingControlService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineMappingControls(body) {
        return this.defineMappingControlService.getDefineMappingControls(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineMappingControl(body) {
        return this.defineMappingControlService.getDefineMappingControl(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineMappingControlController.prototype, "getDefineMappingControls", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineMappingControlController.prototype, "getDefineMappingControl", null);
DefineMappingControlController = __decorate([
    Controller('define-mapping-control'),
    __metadata("design:paramtypes", [DefineMappingControlService])
], DefineMappingControlController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineMappingFunctionService = class DefineMappingFunctionService {
    /**
     * @param {?} defineMappingFunctionRepository
     */
    constructor(defineMappingFunctionRepository) {
        this.defineMappingFunctionRepository = defineMappingFunctionRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineMappingFunction(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineMappingFunctionRepository.find({
                company: context.company,
                pid: context.pid
            });
        });
    }
};
DefineMappingFunctionService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineMappingFunction)),
    __metadata("design:paramtypes", [Repository])
], DefineMappingFunctionService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineMappingFunctionController = class DefineMappingFunctionController {
    /**
     * @param {?} defineMappingFunctionService
     */
    constructor(defineMappingFunctionService) {
        this.defineMappingFunctionService = defineMappingFunctionService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineMappingFunction(body) {
        return this.defineMappingFunctionService.getDefineMappingFunction(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineMappingFunctionController.prototype, "getDefineMappingFunction", null);
DefineMappingFunctionController = __decorate([
    Controller('define-mapping-function'),
    __metadata("design:paramtypes", [DefineMappingFunctionService])
], DefineMappingFunctionController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineMappingService = class DefineMappingService {
    /**
     * @param {?} defineMappingRepository
     */
    constructor(defineMappingRepository) {
        this.defineMappingRepository = defineMappingRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineMapping(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineMappingRepository.find({
                company: context.company,
                pid: context.pid
            });
        });
    }
};
DefineMappingService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineMapping)),
    __metadata("design:paramtypes", [Repository])
], DefineMappingService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineMappingController = class DefineMappingController {
    /**
     * @param {?} defineMappingService
     */
    constructor(defineMappingService) {
        this.defineMappingService = defineMappingService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineMapping(body) {
        return this.defineMappingService.getDefineMapping(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineMappingController.prototype, "getDefineMapping", null);
DefineMappingController = __decorate([
    Controller('define-mapping'),
    __metadata("design:paramtypes", [DefineMappingService])
], DefineMappingController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineSearchService = class DefineSearchService {
    /**
     * @param {?} defineSearchRepository
     */
    constructor(defineSearchRepository) {
        this.defineSearchRepository = defineSearchRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineSearch(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineSearchRepository.find({
                company: context.company,
                pid: context.pid,
                search_id: context.search_id
            });
        });
    }
};
DefineSearchService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineSearch)),
    __metadata("design:paramtypes", [Repository])
], DefineSearchService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineSearchController = class DefineSearchController {
    /**
     * @param {?} defineSearchService
     */
    constructor(defineSearchService) {
        this.defineSearchService = defineSearchService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineSearch(body) {
        return this.defineSearchService.getDefineSearch(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineSearchController.prototype, "getDefineSearch", null);
DefineSearchController = __decorate([
    Controller('define-search'),
    __metadata("design:paramtypes", [DefineSearchService])
], DefineSearchController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DepartmentService = class DepartmentService {
    /**
     * @param {?} departmentRepository
     */
    constructor(departmentRepository) {
        this.departmentRepository = departmentRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDepartments(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.departmentRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDepartment(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.departmentRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
};
DepartmentService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDepartment)),
    __metadata("design:paramtypes", [Repository])
], DepartmentService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DepartmentController = class DepartmentController {
    /**
     * @param {?} departmentService
     */
    constructor(departmentService) {
        this.departmentService = departmentService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDepartments(body) {
        return this.departmentService.getDepartments(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDepartment(body) {
        return this.departmentService.getDepartment(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DepartmentController.prototype, "getDepartments", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DepartmentController.prototype, "getDepartment", null);
DepartmentController = __decorate([
    Controller('department'),
    __metadata("design:paramtypes", [DepartmentService])
], DepartmentController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let EmployeeService = class EmployeeService {
    /**
     * @param {?} employeeRepository
     */
    constructor(employeeRepository) {
        this.employeeRepository = employeeRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getEmployees(context) {
        return __awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let params = new Object();
            /** @type {?} */
            let list = [];
            if (context.name !== undefined) {
                params['name'] = context.name;
            }
            if (context.activeFlag !== undefined) {
                params['active_flag'] = context.activeFlag;
            }
            if (context.address1 !== undefined) {
                params['address1'] = context.address1;
            }
            if (context.address2 !== undefined) {
                params['address2'] = context.address2;
            }
            if (context.tel1 !== undefined) {
                params['tel1'] = context.tel1;
            }
            if (context.tel2 !== undefined) {
                params['tel2'] = context.tel2;
            }
            if (context.email !== undefined) {
                params['email'] = context.email;
            }
            if (context.sortNo !== undefined) {
                params['sort_no'] = context.sortNo;
            }
            if (context.changeCount !== undefined) {
                params['change_count'] = context.changeCount;
            }
            if (context.dataFlag !== undefined) {
                params['data_flag'] = context.dataFlag;
            }
            /** @type {?} */
            let result = yield this.employeeRepository.find(Object.assign({ company: context.company, lang: context.lang }, params));
            list = [...result];
            // NOTE Emp code
            if (context.codeFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.code === context.codeFrom;
                }));
                if (index >= 0) {
                    list = list.slice(index);
                }
            }
            if (context.codeTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.code === context.codeTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE create_emp_id
            if (context.createEmpFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_emp_id === context.createEmpFrom;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.createEmpTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_emp_id === context.createEmpTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE create_datetime
            if (context.createDatetimeFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_datetime === new Date(context.createDatetimeFrom);
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.createDatetimeTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.create_datetime === context.createDatetimeTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE change_emp_id
            if (context.changeEmpFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_emp_id === context.changeEmpFrom;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.changeEmpTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_emp_id === context.changeEmpTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            // NOTE change_datetime
            if (context.changeDatetimeFrom !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_datetime === context.changeDatetimeFrom;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            if (context.changeDatetimeTo !== undefined) {
                /** @type {?} */
                let index = list.findIndex((/**
                 * @param {?} element
                 * @return {?}
                 */
                (element) => {
                    return element.change_datetime === context.changeDatetimeTo;
                }));
                if (index >= 0) {
                    list = list.slice(0, index + 1);
                }
            }
            return list;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getEmployee(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.employeeRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    addEmployee(context) {
        return __awaiter(this, void 0, void 0, function* () {
            if (context.code !== '') {
                /** @type {?} */
                let result = yield this.employeeRepository.find({
                    company: context.company,
                    code: context.code
                });
                if (result.length <= 0) {
                    /** @type {?} */
                    let contextVi = Object.assign({}, context);
                    contextVi.lang = 'vi-VN';
                    /** @type {?} */
                    let contextEn = Object.assign({}, context);
                    contextEn.lang = 'en-US';
                    /** @type {?} */
                    let contextJa = Object.assign({}, context);
                    contextJa.lang = 'ja-JP';
                    yield this.employeeRepository.save(contextVi);
                    yield this.employeeRepository.save(contextEn);
                    yield this.employeeRepository.save(contextJa);
                    return true;
                }
                return false;
            }
            return false;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    updateEmployee(context) {
        return __awaiter(this, void 0, void 0, function* () {
            if (context.code !== '') {
                /** @type {?} */
                let result = yield this.employeeRepository.find({
                    company: context.company,
                    code: context.code
                });
                if (result.length > 0) {
                    context.change_count++;
                    context.change_datetime = null;
                    /** @type {?} */
                    let contextVi = Object.assign({}, context);
                    contextVi.lang = 'vi-VN';
                    /** @type {?} */
                    let contextEn = Object.assign({}, context);
                    contextEn.lang = 'en-US';
                    /** @type {?} */
                    let contextJa = Object.assign({}, context);
                    contextJa.lang = 'ja-JP';
                    yield this.employeeRepository.update([contextVi.company, contextVi.lang, contextVi.code], contextVi);
                    yield this.employeeRepository.update([contextEn.company, contextEn.lang, contextEn.code], contextEn);
                    yield this.employeeRepository.update([contextJa.company, contextJa.lang, contextJa.code], contextJa);
                    return true;
                }
                return false;
            }
            return false;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    deleteEmployee(context) {
        return __awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let result = yield this.employeeRepository.delete({ code: context });
            if (result.affected > 0) {
                return true;
            }
            return false;
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    deleteEmployees(context) {
        return __awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let result = yield this.employeeRepository.delete(context);
            if (result.affected > 0) {
                return true;
            }
            return false;
        });
    }
};
EmployeeService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysEmployee)),
    __metadata("design:paramtypes", [Repository])
], EmployeeService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let EmployeeController = class EmployeeController {
    /**
     * @param {?} employeeService
     */
    constructor(employeeService) {
        this.employeeService = employeeService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getEmployees(body) {
        return this.employeeService.getEmployees(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getEmployee(body) {
        return this.employeeService.getEmployee(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    addEmployee(body) {
        return this.employeeService.addEmployee(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    updateEmployee(body) {
        return this.employeeService.updateEmployee(body);
    }
    /**
     * @param {?} params
     * @return {?}
     */
    deleteEmployee(params) {
        return this.employeeService.deleteEmployee(params.id);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    deleteEmployees(body) {
        return this.employeeService.deleteEmployees(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], EmployeeController.prototype, "getEmployees", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], EmployeeController.prototype, "getEmployee", null);
__decorate([
    Post('add'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], EmployeeController.prototype, "addEmployee", null);
__decorate([
    Post('update'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], EmployeeController.prototype, "updateEmployee", null);
__decorate([
    Delete(':id'),
    __param(0, Param()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], EmployeeController.prototype, "deleteEmployee", null);
__decorate([
    Post('delete/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], EmployeeController.prototype, "deleteEmployees", null);
EmployeeController = __decorate([
    Controller('employee'),
    __metadata("design:paramtypes", [EmployeeService])
], EmployeeController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let LangService = class LangService {
    /**
     * @param {?} langRepository
     */
    constructor(langRepository) {
        this.langRepository = langRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getLangs(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.langRepository.find({
                company: context.company
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getLang(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.langRepository.find({
                company: context.company,
                code: context.code
            });
        });
    }
};
LangService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysLang)),
    __metadata("design:paramtypes", [Repository])
], LangService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let LangController = class LangController {
    /**
     * @param {?} langService
     */
    constructor(langService) {
        this.langService = langService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getLangs(body) {
        return this.langService.getLangs(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getLang(body) {
        return this.langService.getLang(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], LangController.prototype, "getLangs", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], LangController.prototype, "getLang", null);
LangController = __decorate([
    Controller('lang'),
    __metadata("design:paramtypes", [LangService])
], LangController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let MasterDetailService = class MasterDetailService {
    /**
     * @param {?} masterDetailRepository
     */
    constructor(masterDetailRepository) {
        this.masterDetailRepository = masterDetailRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMasterDetails(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.masterDetailRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code,
                master_code: context.master_code
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMasterDetail(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.masterDetailRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code,
                master_code: context.master_code,
                code: context.code
            });
        });
    }
};
MasterDetailService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysMasterDetail)),
    __metadata("design:paramtypes", [Repository])
], MasterDetailService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let MasterDetailController = class MasterDetailController {
    /**
     * @param {?} masterDetailService
     */
    constructor(masterDetailService) {
        this.masterDetailService = masterDetailService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getMasterDetails(body) {
        return this.masterDetailService.getMasterDetails(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getMasterDetail(body) {
        return this.masterDetailService.getMasterDetail(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], MasterDetailController.prototype, "getMasterDetails", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], MasterDetailController.prototype, "getMasterDetail", null);
MasterDetailController = __decorate([
    Controller('master-detail'),
    __metadata("design:paramtypes", [MasterDetailService])
], MasterDetailController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let MasterService = class MasterService {
    /**
     * @param {?} masterRepository
     */
    constructor(masterRepository) {
        this.masterRepository = masterRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMasters(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.masterRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMaster(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.masterRepository.find({
                company: context.company,
                lang: context.lang,
                class_code: context.class_code,
                code: context.code
            });
        });
    }
};
MasterService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysMaster)),
    __metadata("design:paramtypes", [Repository])
], MasterService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let MasterController = class MasterController {
    /**
     * @param {?} masterService
     */
    constructor(masterService) {
        this.masterService = masterService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getMasters(body) {
        return this.masterService.getMasters(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getMaster(body) {
        return this.masterService.getMaster(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], MasterController.prototype, "getMasters", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], MasterController.prototype, "getMaster", null);
MasterController = __decorate([
    Controller('master'),
    __metadata("design:paramtypes", [MasterService])
], MasterController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let MessageService = class MessageService {
    /**
     * @param {?} messageRepository
     */
    constructor(messageRepository) {
        this.messageRepository = messageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getMessage(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.messageRepository.find({
                company: context.company,
                lang: context.lang,
                code: context.code
            });
        });
    }
};
MessageService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysMessage)),
    __metadata("design:paramtypes", [Repository])
], MessageService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let MessageController = class MessageController {
    /**
     * @param {?} messageService
     */
    constructor(messageService) {
        this.messageService = messageService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getMessage(body) {
        return this.messageService.getMessage(body);
    }
};
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], MessageController.prototype, "getMessage", null);
MessageController = __decorate([
    Controller('message'),
    __metadata("design:paramtypes", [MessageService])
], MessageController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let ModuleService = class ModuleService {
    /**
     * @param {?} moduleRepository
     */
    constructor(moduleRepository) {
        this.moduleRepository = moduleRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getModules(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.moduleRepository.find({
                company: context.company,
                lang: context.lang
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getModule(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.moduleRepository.find({
                company: context.company,
                lang: context.lang,
                mid: context.mid
            });
        });
    }
};
ModuleService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysModule)),
    __metadata("design:paramtypes", [Repository])
], ModuleService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let ModuleController = class ModuleController {
    /**
     * @param {?} moduleService
     */
    constructor(moduleService) {
        this.moduleService = moduleService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getModules(body) {
        return this.moduleService.getModules(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getModule(body) {
        return this.moduleService.getModule(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], ModuleController.prototype, "getModules", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], ModuleController.prototype, "getModule", null);
ModuleController = __decorate([
    Controller('module'),
    __metadata("design:paramtypes", [ModuleService])
], ModuleController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let PageService = class PageService {
    /**
     * @param {?} pageRepository
     */
    constructor(pageRepository) {
        this.pageRepository = pageRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getPages(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.pageRepository.find({
                company: context.company,
                lang: context.lang,
                mid: context.mid
            });
        });
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getPage(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.pageRepository.find({
                company: context.company,
                lang: context.lang,
                mid: context.mid,
                pid: context.pid
            });
        });
    }
};
PageService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysPage)),
    __metadata("design:paramtypes", [Repository])
], PageService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let PageController = class PageController {
    /**
     * @param {?} pageService
     */
    constructor(pageService) {
        this.pageService = pageService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getPages(body) {
        return this.pageService.getPages(body);
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getPage(body) {
        return this.pageService.getPage(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], PageController.prototype, "getPages", null);
__decorate([
    Post('get/one'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], PageController.prototype, "getPage", null);
PageController = __decorate([
    Controller('page'),
    __metadata("design:paramtypes", [PageService])
], PageController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineInputGridService = class DefineInputGridService {
    /**
     * @param {?} defineInputGridRepository
     */
    constructor(defineInputGridRepository) {
        this.defineInputGridRepository = defineInputGridRepository;
    }
    /**
     * @param {?} context
     * @return {?}
     */
    getDefineInputGrid(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.defineInputGridRepository.find({
                company: context.company,
                pid: context.pid,
                input_grid_id: context.input_grid_id
            });
        });
    }
};
DefineInputGridService = __decorate([
    Injectable(),
    __param(0, InjectRepository(SysDefineInputGrid)),
    __metadata("design:paramtypes", [Repository])
], DefineInputGridService);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
let DefineInputGridController = class DefineInputGridController {
    /**
     * @param {?} defineInputGridService
     */
    constructor(defineInputGridService) {
        this.defineInputGridService = defineInputGridService;
    }
    /**
     * @param {?} body
     * @return {?}
     */
    getDefineInputGrid(body) {
        return this.defineInputGridService.getDefineInputGrid(body);
    }
};
__decorate([
    Post('get/many'),
    __param(0, Body()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], DefineInputGridController.prototype, "getDefineInputGrid", null);
DefineInputGridController = __decorate([
    Controller('define-input-grid'),
    __metadata("design:paramtypes", [DefineInputGridService])
], DefineInputGridController);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
const controllers = [
    CaptionController,
    CompanyController,
    DefineClassController,
    DefineDisplayController,
    DefineGridController,
    DefineGroupController,
    DefineInputGridController,
    DefineInputController,
    DefineMappingControlController,
    DefineMappingFunctionController,
    DefineMappingController,
    DefineSearchController,
    DepartmentController,
    EmployeeController,
    LangController,
    MasterDetailController,
    MasterController,
    MessageController,
    ModuleController,
    PageController
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
const services = [
    CaptionService,
    CompanyService,
    DefineClassService,
    DefineDisplayService,
    DefineGridService,
    DefineGroupService,
    DefineInputGridService,
    DefineInputService,
    DefineMappingControlService,
    DefineMappingFunctionService,
    DefineMappingService,
    DefineSearchService,
    DepartmentService,
    EmployeeService,
    LangService,
    MasterDetailService,
    MasterService,
    MessageService,
    ModuleService,
    PageService
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AitCoreModule {
}
AitCoreModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    RouterModule.forChild([
                    /* {path: '', pathMatch: 'full', component: InsertYourComponentHere} */
                    ])
                ]
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { AitCoreModule, controllers, dbConfig, entities, services, SysCaption as a, SysCompany as b, DefineDisplayController as ba, DefineDisplayService as bb, DefineGridController as bc, DefineGridService as bd, DefineGroupController as be, DefineGroupService as bf, DefineInputGridController as bg, DefineInputGridService as bh, DefineInputController as bi, DefineInputService as bj, DefineMappingControlController as bk, DefineMappingControlService as bl, DefineMappingFunctionController as bm, DefineMappingFunctionService as bn, DefineMappingController as bo, DefineMappingService as bp, DefineSearchController as bq, DefineSearchService as br, DepartmentController as bs, DepartmentService as bt, EmployeeController as bu, EmployeeService as bv, LangController as bw, LangService as bx, MasterDetailController as by, MasterDetailService as bz, SysDefineClass as c, MasterController as ca, MasterService as cb, MessageController as cc, MessageService as cd, ModuleController as ce, ModuleService as cf, PageController as cg, PageService as ch, SysDefineDisplay as d, SysDefineGrid as e, SysDefineGroup as f, SysDefineInput as g, SysDefineInputGrid as h, SysDefineMapping as i, SysDefineMappingControl as j, SysDefineMappingFunction as k, SysDefineSearch as l, SysDepartment as m, SysEmployee as n, SysLang as o, SysMaster as p, SysMasterDetail as q, SysMessage as r, SysModule as s, SysPage as t, CaptionController as u, CaptionService as v, CompanyController as w, CompanyService as x, DefineClassController as y, DefineClassService as z };
